{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from malid import config, helpers\n",
    "from malid.datamodels import healthy_label"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option(\"display.max_rows\", 500)\n",
    "pd.set_option(\"display.max_columns\", 500)\n",
    "pd.set_option(\"display.width\", 1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# produce metadata about external cohorts\n",
    "\n",
    "produces:\n",
    "\n",
    "- all cohorts\n",
    "    - `metadata/generated.external_cohorts.all_bcr.participant_metadata.tsv`\n",
    "- covid\n",
    "    - `metadata/generated.external_cohorts.covid19_bcr.specimen_metadata_extra.tsv`\n",
    "    - `metadata/generated.external_cohorts.covid19_bcr.participant_metadata.tsv`\n",
    "- healthy\n",
    "    - `metadata/generated.external_cohorts.healthy_bcr.participant_metadata.tsv`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# external covid cohorts\n",
    "\n",
    "## load metadata\n",
    "\n",
    "`repertoire_id`'s are like specimen IDs. they are many-to-one with patient IDs. below, we will introduce actual patient IDs for these external cohorts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(184, 103)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(\n",
    "    config.paths.external_raw_data / \"Kim\" / \"airr_covid19_metadata.tsv\",\n",
    "    sep=\"\\t\",\n",
    ")\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>repertoire_name</th>\n",
       "      <th>repertoire_description</th>\n",
       "      <th>study.study_id</th>\n",
       "      <th>study.study_title</th>\n",
       "      <th>study.study_type.label</th>\n",
       "      <th>study.study_type.id</th>\n",
       "      <th>study.study_description</th>\n",
       "      <th>study.inclusion_exclusion_criteria</th>\n",
       "      <th>study.grants</th>\n",
       "      <th>study.collected_by</th>\n",
       "      <th>study.lab_name</th>\n",
       "      <th>study.lab_address</th>\n",
       "      <th>study.submitted_by</th>\n",
       "      <th>study.pub_ids</th>\n",
       "      <th>study.keywords_study.0</th>\n",
       "      <th>subject.subject_id</th>\n",
       "      <th>subject.synthetic</th>\n",
       "      <th>subject.species.label</th>\n",
       "      <th>subject.species.id</th>\n",
       "      <th>subject.organism.label</th>\n",
       "      <th>subject.organism.id</th>\n",
       "      <th>subject.sex</th>\n",
       "      <th>subject.age</th>\n",
       "      <th>subject.age_min</th>\n",
       "      <th>subject.age_max</th>\n",
       "      <th>subject.age_unit.label</th>\n",
       "      <th>subject.age_unit.id</th>\n",
       "      <th>subject.age_event</th>\n",
       "      <th>subject.ancestry_population</th>\n",
       "      <th>subject.ethnicity</th>\n",
       "      <th>subject.race</th>\n",
       "      <th>subject.strain_name</th>\n",
       "      <th>subject.linked_subjects</th>\n",
       "      <th>subject.link_type</th>\n",
       "      <th>subject.diagnosis.0.study_group_description</th>\n",
       "      <th>subject.diagnosis.0.disease_diagnosis.label</th>\n",
       "      <th>subject.diagnosis.0.disease_diagnosis.id</th>\n",
       "      <th>subject.diagnosis.0.disease_length</th>\n",
       "      <th>subject.diagnosis.0.disease_stage</th>\n",
       "      <th>subject.diagnosis.0.prior_therapies</th>\n",
       "      <th>subject.diagnosis.0.immunogen</th>\n",
       "      <th>subject.diagnosis.0.intervention</th>\n",
       "      <th>subject.diagnosis.0.medical_history</th>\n",
       "      <th>sample.0.sample_processing_id</th>\n",
       "      <th>sample.0.sample_id</th>\n",
       "      <th>sample.0.sample_type</th>\n",
       "      <th>sample.0.tissue.label</th>\n",
       "      <th>sample.0.tissue.id</th>\n",
       "      <th>sample.0.anatomic_site</th>\n",
       "      <th>sample.0.disease_state_sample</th>\n",
       "      <th>sample.0.collection_time_point_relative</th>\n",
       "      <th>sample.0.collection_time_point_reference</th>\n",
       "      <th>sample.0.biomaterial_provider</th>\n",
       "      <th>sample.0.tissue_processing</th>\n",
       "      <th>sample.0.cell_subset.label</th>\n",
       "      <th>sample.0.cell_subset.id</th>\n",
       "      <th>sample.0.cell_phenotype</th>\n",
       "      <th>sample.0.cell_species.label</th>\n",
       "      <th>sample.0.cell_species.id</th>\n",
       "      <th>sample.0.single_cell</th>\n",
       "      <th>sample.0.cell_number</th>\n",
       "      <th>sample.0.cells_per_reaction</th>\n",
       "      <th>sample.0.cell_storage</th>\n",
       "      <th>sample.0.cell_quality</th>\n",
       "      <th>sample.0.cell_isolation</th>\n",
       "      <th>sample.0.cell_processing_protocol</th>\n",
       "      <th>sample.0.template_class</th>\n",
       "      <th>sample.0.template_quality</th>\n",
       "      <th>sample.0.template_amount</th>\n",
       "      <th>sample.0.library_generation_method</th>\n",
       "      <th>sample.0.library_generation_protocol</th>\n",
       "      <th>sample.0.library_generation_kit_version</th>\n",
       "      <th>sample.0.pcr_target.0.pcr_target_locus</th>\n",
       "      <th>sample.0.pcr_target.0.forward_pcr_primer_target_location</th>\n",
       "      <th>sample.0.pcr_target.0.reverse_pcr_primer_target_location</th>\n",
       "      <th>sample.0.complete_sequences</th>\n",
       "      <th>sample.0.physical_linkage</th>\n",
       "      <th>sample.0.sequencing_run_id</th>\n",
       "      <th>sample.0.total_reads_passing_qc_filter</th>\n",
       "      <th>sample.0.sequencing_platform</th>\n",
       "      <th>sample.0.sequencing_facility</th>\n",
       "      <th>sample.0.sequencing_run_date</th>\n",
       "      <th>sample.0.sequencing_kit</th>\n",
       "      <th>sample.0.sequencing_files.file_type</th>\n",
       "      <th>sample.0.sequencing_files.filename</th>\n",
       "      <th>sample.0.sequencing_files.read_direction</th>\n",
       "      <th>sample.0.sequencing_files.read_length</th>\n",
       "      <th>sample.0.sequencing_files.paired_filename</th>\n",
       "      <th>sample.0.sequencing_files.paired_read_direction</th>\n",
       "      <th>sample.0.sequencing_files.paired_read_length</th>\n",
       "      <th>data_processing.0.data_processing_id</th>\n",
       "      <th>data_processing.0.primary_annotation</th>\n",
       "      <th>data_processing.0.software_versions</th>\n",
       "      <th>data_processing.0.paired_reads_assembly</th>\n",
       "      <th>data_processing.0.quality_thresholds</th>\n",
       "      <th>data_processing.0.primer_match_cutoffs</th>\n",
       "      <th>data_processing.0.collapsing_method</th>\n",
       "      <th>data_processing.0.data_processing_protocols</th>\n",
       "      <th>data_processing.0.data_processing_files.0</th>\n",
       "      <th>data_processing.0.germline_database</th>\n",
       "      <th>data_processing.0.analysis_provenance_id</th>\n",
       "      <th>study.keywords_study.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5ed6859e99011334ac05e847</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>B cell clonal expansion and convergent antibod...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>COVID-19 Study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...</td>\n",
       "      <td>S. Boyd, sboyd1@stanford.edu</td>\n",
       "      <td>Scott Boyd Lab, Department of Pathology</td>\n",
       "      <td>Stanford University</td>\n",
       "      <td>S. Nielsen, F. Yang, S. Boyd</td>\n",
       "      <td>DOI:10.21203/rs.3.rs-27220/v1</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>7450</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>73.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COVID_active</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed6859e99011334ac05e847</td>\n",
       "      <td>M369-S008</td>\n",
       "      <td>Venipuncture blood samples were collected in K...</td>\n",
       "      <td>venous blood</td>\n",
       "      <td>UBERON:0013756</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "      <td>Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>355342.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The AllPrep DNA/RNA kit (Qiagen) was used to e...</td>\n",
       "      <td>DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100 ng/library</td>\n",
       "      <td>PCR</td>\n",
       "      <td>For each blood sample, six independent gDNA li...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV_FR1 or IGHV_FR2</td>\n",
       "      <td>IGHJ</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>289023.0</td>\n",
       "      <td>Illumina MiSeq (PE300)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta</td>\n",
       "      <td>SRR11610494.fasta</td>\n",
       "      <td>NaN</td>\n",
       "      <td>317.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed6859e99011334ac05e847</td>\n",
       "      <td>True</td>\n",
       "      <td>fastx-toolkit/0.0.14, IgBLAST/1.16.0</td>\n",
       "      <td>FLASH20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adaptors and primers removed by authors. Igbla...</td>\n",
       "      <td>SRR11610494.fasta.fmt19-filtered.fmt19</td>\n",
       "      <td>IMGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5ed685a099011334ac05e848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>B cell clonal expansion and convergent antibod...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>COVID-19 Study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...</td>\n",
       "      <td>S. Boyd, sboyd1@stanford.edu</td>\n",
       "      <td>Scott Boyd Lab, Department of Pathology</td>\n",
       "      <td>Stanford University</td>\n",
       "      <td>S. Nielsen, F. Yang, S. Boyd</td>\n",
       "      <td>DOI:10.21203/rs.3.rs-27220/v1</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>7451</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>62.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COVID_active</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed685a099011334ac05e848</td>\n",
       "      <td>M369-S009</td>\n",
       "      <td>Venipuncture blood samples were collected in K...</td>\n",
       "      <td>venous blood</td>\n",
       "      <td>UBERON:0013756</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "      <td>Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>347058.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The AllPrep DNA/RNA kit (Qiagen) was used to e...</td>\n",
       "      <td>DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100 ng/library</td>\n",
       "      <td>PCR</td>\n",
       "      <td>For each blood sample, six independent gDNA li...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV_FR1 or IGHV_FR2</td>\n",
       "      <td>IGHJ</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>279041.0</td>\n",
       "      <td>Illumina MiSeq (PE300)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta</td>\n",
       "      <td>SRR11610493.fasta</td>\n",
       "      <td>NaN</td>\n",
       "      <td>317.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed685a099011334ac05e848</td>\n",
       "      <td>True</td>\n",
       "      <td>fastx-toolkit/0.0.14, IgBLAST/1.16.0</td>\n",
       "      <td>FLASH20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adaptors and primers removed by authors. Igbla...</td>\n",
       "      <td>SRR11610493.fasta.fmt19-filtered.fmt19</td>\n",
       "      <td>IMGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5ed685a099011334ac05e849</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>B cell clonal expansion and convergent antibod...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>COVID-19 Study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...</td>\n",
       "      <td>S. Boyd, sboyd1@stanford.edu</td>\n",
       "      <td>Scott Boyd Lab, Department of Pathology</td>\n",
       "      <td>Stanford University</td>\n",
       "      <td>S. Nielsen, F. Yang, S. Boyd</td>\n",
       "      <td>DOI:10.21203/rs.3.rs-27220/v1</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>7452</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>61.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COVID_active</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed685a099011334ac05e849</td>\n",
       "      <td>M369-S010</td>\n",
       "      <td>Venipuncture blood samples were collected in K...</td>\n",
       "      <td>venous blood</td>\n",
       "      <td>UBERON:0013756</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "      <td>Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>365783.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The AllPrep DNA/RNA kit (Qiagen) was used to e...</td>\n",
       "      <td>DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100 ng/library</td>\n",
       "      <td>PCR</td>\n",
       "      <td>For each blood sample, six independent gDNA li...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV_FR1 or IGHV_FR2</td>\n",
       "      <td>IGHJ</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>283204.0</td>\n",
       "      <td>Illumina MiSeq (PE300)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta</td>\n",
       "      <td>SRR11610492.fasta</td>\n",
       "      <td>NaN</td>\n",
       "      <td>317.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed685a099011334ac05e849</td>\n",
       "      <td>True</td>\n",
       "      <td>fastx-toolkit/0.0.14, IgBLAST/1.16.0</td>\n",
       "      <td>FLASH20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adaptors and primers removed by authors. Igbla...</td>\n",
       "      <td>SRR11610492.fasta.fmt19-filtered.fmt19</td>\n",
       "      <td>IMGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5ed685a099011334ac05e84a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>B cell clonal expansion and convergent antibod...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>COVID-19 Study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...</td>\n",
       "      <td>S. Boyd, sboyd1@stanford.edu</td>\n",
       "      <td>Scott Boyd Lab, Department of Pathology</td>\n",
       "      <td>Stanford University</td>\n",
       "      <td>S. Nielsen, F. Yang, S. Boyd</td>\n",
       "      <td>DOI:10.21203/rs.3.rs-27220/v1</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>7453-D0</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COVID_active</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed685a099011334ac05e84a</td>\n",
       "      <td>M369-S011</td>\n",
       "      <td>Venipuncture blood samples were collected in K...</td>\n",
       "      <td>venous blood</td>\n",
       "      <td>UBERON:0013756</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "      <td>Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>349566.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The AllPrep DNA/RNA kit (Qiagen) was used to e...</td>\n",
       "      <td>DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25 ng/library</td>\n",
       "      <td>PCR</td>\n",
       "      <td>For each blood sample, six independent gDNA li...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV_FR1 or IGHV_FR2</td>\n",
       "      <td>IGHJ</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>276877.0</td>\n",
       "      <td>Illumina MiSeq (PE300)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta</td>\n",
       "      <td>SRR11610503.fasta</td>\n",
       "      <td>NaN</td>\n",
       "      <td>317.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed685a099011334ac05e84a</td>\n",
       "      <td>True</td>\n",
       "      <td>fastx-toolkit/0.0.14, IgBLAST/1.16.0</td>\n",
       "      <td>FLASH20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adaptors and primers removed by authors. Igbla...</td>\n",
       "      <td>SRR11610503.fasta.fmt19-filtered.fmt19</td>\n",
       "      <td>IMGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5ed685a099011334ac05e84b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>B cell clonal expansion and convergent antibod...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>COVID-19 Study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...</td>\n",
       "      <td>S. Boyd, sboyd1@stanford.edu</td>\n",
       "      <td>Scott Boyd Lab, Department of Pathology</td>\n",
       "      <td>Stanford University</td>\n",
       "      <td>S. Nielsen, F. Yang, S. Boyd</td>\n",
       "      <td>DOI:10.21203/rs.3.rs-27220/v1</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>7453-D2</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COVID_active</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed685a099011334ac05e84b</td>\n",
       "      <td>M369-S012</td>\n",
       "      <td>Venipuncture blood samples were collected in K...</td>\n",
       "      <td>venous blood</td>\n",
       "      <td>UBERON:0013756</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "      <td>Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>398829.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The AllPrep DNA/RNA kit (Qiagen) was used to e...</td>\n",
       "      <td>DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100 ng/library</td>\n",
       "      <td>PCR</td>\n",
       "      <td>For each blood sample, six independent gDNA li...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV_FR1 or IGHV_FR2</td>\n",
       "      <td>IGHJ</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>321208.0</td>\n",
       "      <td>Illumina MiSeq (PE300)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta</td>\n",
       "      <td>SRR11610502.fasta</td>\n",
       "      <td>NaN</td>\n",
       "      <td>317.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5ed685a099011334ac05e84b</td>\n",
       "      <td>True</td>\n",
       "      <td>fastx-toolkit/0.0.14, IgBLAST/1.16.0</td>\n",
       "      <td>FLASH20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adaptors and primers removed by authors. Igbla...</td>\n",
       "      <td>SRR11610502.fasta.fmt19-filtered.fmt19</td>\n",
       "      <td>IMGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              repertoire_id  repertoire_name  repertoire_description study.study_id                                  study.study_title study.study_type.label study.study_type.id study.study_description study.inclusion_exclusion_criteria                                       study.grants            study.collected_by                           study.lab_name    study.lab_address            study.submitted_by                  study.pub_ids study.keywords_study.0 subject.subject_id  subject.synthetic subject.species.label subject.species.id  subject.organism.label  subject.organism.id subject.sex  subject.age  subject.age_min  subject.age_max subject.age_unit.label subject.age_unit.id  subject.age_event  subject.ancestry_population subject.ethnicity subject.race  subject.strain_name  subject.linked_subjects  subject.link_type subject.diagnosis.0.study_group_description subject.diagnosis.0.disease_diagnosis.label subject.diagnosis.0.disease_diagnosis.id subject.diagnosis.0.disease_length  \\\n",
       "0  5ed6859e99011334ac05e847              NaN                     NaN    PRJNA628125  B cell clonal expansion and convergent antibod...     Case-Control Study         NCIT:C15197          COVID-19 Study                                NaN  NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...  S. Boyd, sboyd1@stanford.edu  Scott Boyd Lab, Department of Pathology  Stanford University  S. Nielsen, F. Yang, S. Boyd  DOI:10.21203/rs.3.rs-27220/v1            contains_ig               7450              False          Homo sapiens     NCBITAXON:9606                     NaN                  NaN      female          NaN             73.0             73.0                   year          UO:0000036  Sample Collection                          NaN               NaN          NaN                  NaN                      NaN                NaN                                        Case                                    COVID-19                             DOID:0080600                                NaN   \n",
       "1  5ed685a099011334ac05e848              NaN                     NaN    PRJNA628125  B cell clonal expansion and convergent antibod...     Case-Control Study         NCIT:C15197          COVID-19 Study                                NaN  NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...  S. Boyd, sboyd1@stanford.edu  Scott Boyd Lab, Department of Pathology  Stanford University  S. Nielsen, F. Yang, S. Boyd  DOI:10.21203/rs.3.rs-27220/v1            contains_ig               7451              False          Homo sapiens     NCBITAXON:9606                     NaN                  NaN        male          NaN             62.0             62.0                   year          UO:0000036  Sample Collection                          NaN               NaN          NaN                  NaN                      NaN                NaN                                        Case                                    COVID-19                             DOID:0080600                                NaN   \n",
       "2  5ed685a099011334ac05e849              NaN                     NaN    PRJNA628125  B cell clonal expansion and convergent antibod...     Case-Control Study         NCIT:C15197          COVID-19 Study                                NaN  NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...  S. Boyd, sboyd1@stanford.edu  Scott Boyd Lab, Department of Pathology  Stanford University  S. Nielsen, F. Yang, S. Boyd  DOI:10.21203/rs.3.rs-27220/v1            contains_ig               7452              False          Homo sapiens     NCBITAXON:9606                     NaN                  NaN      female          NaN             61.0             61.0                   year          UO:0000036  Sample Collection                          NaN               NaN          NaN                  NaN                      NaN                NaN                                        Case                                    COVID-19                             DOID:0080600                                NaN   \n",
       "3  5ed685a099011334ac05e84a              NaN                     NaN    PRJNA628125  B cell clonal expansion and convergent antibod...     Case-Control Study         NCIT:C15197          COVID-19 Study                                NaN  NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...  S. Boyd, sboyd1@stanford.edu  Scott Boyd Lab, Department of Pathology  Stanford University  S. Nielsen, F. Yang, S. Boyd  DOI:10.21203/rs.3.rs-27220/v1            contains_ig            7453-D0              False          Homo sapiens     NCBITAXON:9606                     NaN                  NaN        male          NaN             64.0             64.0                   year          UO:0000036  Sample Collection                          NaN               NaN          NaN                  NaN                      NaN                NaN                                        Case                                    COVID-19                             DOID:0080600                                NaN   \n",
       "4  5ed685a099011334ac05e84b              NaN                     NaN    PRJNA628125  B cell clonal expansion and convergent antibod...     Case-Control Study         NCIT:C15197          COVID-19 Study                                NaN  NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...  S. Boyd, sboyd1@stanford.edu  Scott Boyd Lab, Department of Pathology  Stanford University  S. Nielsen, F. Yang, S. Boyd  DOI:10.21203/rs.3.rs-27220/v1            contains_ig            7453-D2              False          Homo sapiens     NCBITAXON:9606                     NaN                  NaN        male          NaN             64.0             64.0                   year          UO:0000036  Sample Collection                          NaN               NaN          NaN                  NaN                      NaN                NaN                                        Case                                    COVID-19                             DOID:0080600                                NaN   \n",
       "\n",
       "  subject.diagnosis.0.disease_stage subject.diagnosis.0.prior_therapies  subject.diagnosis.0.immunogen  subject.diagnosis.0.intervention subject.diagnosis.0.medical_history sample.0.sample_processing_id sample.0.sample_id                               sample.0.sample_type sample.0.tissue.label sample.0.tissue.id  sample.0.anatomic_site  sample.0.disease_state_sample sample.0.collection_time_point_relative sample.0.collection_time_point_reference                      sample.0.biomaterial_provider sample.0.tissue_processing  sample.0.cell_subset.label  sample.0.cell_subset.id  sample.0.cell_phenotype sample.0.cell_species.label sample.0.cell_species.id  sample.0.single_cell  sample.0.cell_number  sample.0.cells_per_reaction sample.0.cell_storage sample.0.cell_quality  sample.0.cell_isolation                  sample.0.cell_processing_protocol sample.0.template_class  sample.0.template_quality sample.0.template_amount sample.0.library_generation_method  \\\n",
       "0                      COVID_active                                 NaN                            NaN                               NaN                                 NaN      5ed6859e99011334ac05e847          M369-S008  Venipuncture blood samples were collected in K...          venous blood     UBERON:0013756                     NaN                            NaN                                     0 d                       Hospital Admission  Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...                        NaN                         NaN                      NaN                      NaN                Homo sapiens           NCBITAXON:9606                 False              355342.0                          NaN                 False                   NaN                      NaN  The AllPrep DNA/RNA kit (Qiagen) was used to e...                     DNA                        NaN           100 ng/library                                PCR   \n",
       "1                      COVID_active                                 NaN                            NaN                               NaN                                 NaN      5ed685a099011334ac05e848          M369-S009  Venipuncture blood samples were collected in K...          venous blood     UBERON:0013756                     NaN                            NaN                                     0 d                       Hospital Admission  Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...                        NaN                         NaN                      NaN                      NaN                Homo sapiens           NCBITAXON:9606                 False              347058.0                          NaN                 False                   NaN                      NaN  The AllPrep DNA/RNA kit (Qiagen) was used to e...                     DNA                        NaN           100 ng/library                                PCR   \n",
       "2                      COVID_active                                 NaN                            NaN                               NaN                                 NaN      5ed685a099011334ac05e849          M369-S010  Venipuncture blood samples were collected in K...          venous blood     UBERON:0013756                     NaN                            NaN                                     0 d                       Hospital Admission  Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...                        NaN                         NaN                      NaN                      NaN                Homo sapiens           NCBITAXON:9606                 False              365783.0                          NaN                 False                   NaN                      NaN  The AllPrep DNA/RNA kit (Qiagen) was used to e...                     DNA                        NaN           100 ng/library                                PCR   \n",
       "3                      COVID_active                                 NaN                            NaN                               NaN                                 NaN      5ed685a099011334ac05e84a          M369-S011  Venipuncture blood samples were collected in K...          venous blood     UBERON:0013756                     NaN                            NaN                                     0 d                       Hospital Admission  Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...                        NaN                         NaN                      NaN                      NaN                Homo sapiens           NCBITAXON:9606                 False              349566.0                          NaN                 False                   NaN                      NaN  The AllPrep DNA/RNA kit (Qiagen) was used to e...                     DNA                        NaN            25 ng/library                                PCR   \n",
       "4                      COVID_active                                 NaN                            NaN                               NaN                                 NaN      5ed685a099011334ac05e84b          M369-S012  Venipuncture blood samples were collected in K...          venous blood     UBERON:0013756                     NaN                            NaN                                     2 d                       Hospital Admission  Scott Boyd lab, R214 Edwards, 300 Pasteur Driv...                        NaN                         NaN                      NaN                      NaN                Homo sapiens           NCBITAXON:9606                 False              398829.0                          NaN                 False                   NaN                      NaN  The AllPrep DNA/RNA kit (Qiagen) was used to e...                     DNA                        NaN           100 ng/library                                PCR   \n",
       "\n",
       "                sample.0.library_generation_protocol sample.0.library_generation_kit_version sample.0.pcr_target.0.pcr_target_locus sample.0.pcr_target.0.forward_pcr_primer_target_location sample.0.pcr_target.0.reverse_pcr_primer_target_location sample.0.complete_sequences sample.0.physical_linkage  sample.0.sequencing_run_id  sample.0.total_reads_passing_qc_filter sample.0.sequencing_platform sample.0.sequencing_facility  sample.0.sequencing_run_date sample.0.sequencing_kit sample.0.sequencing_files.file_type sample.0.sequencing_files.filename  sample.0.sequencing_files.read_direction  sample.0.sequencing_files.read_length sample.0.sequencing_files.paired_filename  sample.0.sequencing_files.paired_read_direction  sample.0.sequencing_files.paired_read_length data_processing.0.data_processing_id  data_processing.0.primary_annotation   data_processing.0.software_versions data_processing.0.paired_reads_assembly data_processing.0.quality_thresholds data_processing.0.primer_match_cutoffs  \\\n",
       "0  For each blood sample, six independent gDNA li...                                     NaN                                    IGH                               IGHV_FR1 or IGHV_FR2                                                     IGHJ                           partial                      none                         NaN                                289023.0       Illumina MiSeq (PE300)                          NaN                           NaN                     NaN                               fasta                  SRR11610494.fasta                                       NaN                                  317.0                                       NaN                                              NaN                                           NaN             5ed6859e99011334ac05e847                                  True  fastx-toolkit/0.0.14, IgBLAST/1.16.0                                 FLASH20                                  NaN                                    NaN   \n",
       "1  For each blood sample, six independent gDNA li...                                     NaN                                    IGH                               IGHV_FR1 or IGHV_FR2                                                     IGHJ                           partial                      none                         NaN                                279041.0       Illumina MiSeq (PE300)                          NaN                           NaN                     NaN                               fasta                  SRR11610493.fasta                                       NaN                                  317.0                                       NaN                                              NaN                                           NaN             5ed685a099011334ac05e848                                  True  fastx-toolkit/0.0.14, IgBLAST/1.16.0                                 FLASH20                                  NaN                                    NaN   \n",
       "2  For each blood sample, six independent gDNA li...                                     NaN                                    IGH                               IGHV_FR1 or IGHV_FR2                                                     IGHJ                           partial                      none                         NaN                                283204.0       Illumina MiSeq (PE300)                          NaN                           NaN                     NaN                               fasta                  SRR11610492.fasta                                       NaN                                  317.0                                       NaN                                              NaN                                           NaN             5ed685a099011334ac05e849                                  True  fastx-toolkit/0.0.14, IgBLAST/1.16.0                                 FLASH20                                  NaN                                    NaN   \n",
       "3  For each blood sample, six independent gDNA li...                                     NaN                                    IGH                               IGHV_FR1 or IGHV_FR2                                                     IGHJ                           partial                      none                         NaN                                276877.0       Illumina MiSeq (PE300)                          NaN                           NaN                     NaN                               fasta                  SRR11610503.fasta                                       NaN                                  317.0                                       NaN                                              NaN                                           NaN             5ed685a099011334ac05e84a                                  True  fastx-toolkit/0.0.14, IgBLAST/1.16.0                                 FLASH20                                  NaN                                    NaN   \n",
       "4  For each blood sample, six independent gDNA li...                                     NaN                                    IGH                               IGHV_FR1 or IGHV_FR2                                                     IGHJ                           partial                      none                         NaN                                321208.0       Illumina MiSeq (PE300)                          NaN                           NaN                     NaN                               fasta                  SRR11610502.fasta                                       NaN                                  317.0                                       NaN                                              NaN                                           NaN             5ed685a099011334ac05e84b                                  True  fastx-toolkit/0.0.14, IgBLAST/1.16.0                                 FLASH20                                  NaN                                    NaN   \n",
       "\n",
       "  data_processing.0.collapsing_method        data_processing.0.data_processing_protocols data_processing.0.data_processing_files.0 data_processing.0.germline_database  data_processing.0.analysis_provenance_id study.keywords_study.1  \n",
       "0                                 NaN  Adaptors and primers removed by authors. Igbla...    SRR11610494.fasta.fmt19-filtered.fmt19                                IMGT                                       NaN                    NaN  \n",
       "1                                 NaN  Adaptors and primers removed by authors. Igbla...    SRR11610493.fasta.fmt19-filtered.fmt19                                IMGT                                       NaN                    NaN  \n",
       "2                                 NaN  Adaptors and primers removed by authors. Igbla...    SRR11610492.fasta.fmt19-filtered.fmt19                                IMGT                                       NaN                    NaN  \n",
       "3                                 NaN  Adaptors and primers removed by authors. Igbla...    SRR11610503.fasta.fmt19-filtered.fmt19                                IMGT                                       NaN                    NaN  \n",
       "4                                 NaN  Adaptors and primers removed by authors. Igbla...    SRR11610502.fasta.fmt19-filtered.fmt19                                IMGT                                       NaN                    NaN  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study.study_id</th>\n",
       "      <th>subject.subject_id</th>\n",
       "      <th>sample.0.collection_time_point_relative</th>\n",
       "      <th>sample.0.collection_time_point_reference</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5ed6859e99011334ac05e847</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>7450</td>\n",
       "      <td>0 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5ed685a099011334ac05e848</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>7451</td>\n",
       "      <td>0 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5ed685a099011334ac05e849</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>7452</td>\n",
       "      <td>0 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5ed685a099011334ac05e84a</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>7453-D0</td>\n",
       "      <td>0 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5ed685a099011334ac05e84b</td>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>7453-D2</td>\n",
       "      <td>2 d</td>\n",
       "      <td>Hospital Admission</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              repertoire_id study.study_id subject.subject_id sample.0.collection_time_point_relative sample.0.collection_time_point_reference\n",
       "0  5ed6859e99011334ac05e847    PRJNA628125               7450                                     0 d                       Hospital Admission\n",
       "1  5ed685a099011334ac05e848    PRJNA628125               7451                                     0 d                       Hospital Admission\n",
       "2  5ed685a099011334ac05e849    PRJNA628125               7452                                     0 d                       Hospital Admission\n",
       "3  5ed685a099011334ac05e84a    PRJNA628125            7453-D0                                     0 d                       Hospital Admission\n",
       "4  5ed685a099011334ac05e84b    PRJNA628125            7453-D2                                     2 d                       Hospital Admission"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\n",
    "    [\n",
    "        \"repertoire_id\",\n",
    "        \"study.study_id\",\n",
    "        \"subject.subject_id\",\n",
    "        \"sample.0.collection_time_point_relative\",\n",
    "        \"sample.0.collection_time_point_reference\",\n",
    "    ]\n",
    "].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Sample collection     19\n",
       "0 d                   12\n",
       "14 d                  12\n",
       "8 d                    6\n",
       "25 d                   6\n",
       "7 d                    6\n",
       "17 d                   6\n",
       "13 d                   6\n",
       "5 d                    6\n",
       "16 d                   6\n",
       "d20                    5\n",
       "Less than 7 days       5\n",
       "d39                    5\n",
       "More than 14 days      5\n",
       "d23                    4\n",
       "1 d                    4\n",
       "d11                    3\n",
       "d16                    3\n",
       "d45                    3\n",
       "d6                     3\n",
       "d37                    3\n",
       "d9                     3\n",
       "d15                    3\n",
       "d36                    2\n",
       "d46                    2\n",
       "d51                    2\n",
       "d14                    2\n",
       "d28                    2\n",
       "d18                    2\n",
       "d22                    2\n",
       "d44                    2\n",
       "2 d                    2\n",
       "d1                     2\n",
       "d5                     2\n",
       "d24                    2\n",
       "d31                    2\n",
       "d27                    2\n",
       "d38                    2\n",
       "d40                    1\n",
       "d19                    1\n",
       "d33                    1\n",
       "d2                     1\n",
       "d25                    1\n",
       "d99                    1\n",
       "d26                    1\n",
       "d10                    1\n",
       "d17                    1\n",
       "d41                    1\n",
       "d35                    1\n",
       "d32                    1\n",
       "d13                    1\n",
       "d50                    1\n",
       "d47                    1\n",
       "d34                    1\n",
       "d21                    1\n",
       "d29                    1\n",
       "d8                     1\n",
       "d52                    1\n",
       "Name: sample.0.collection_time_point_relative, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"sample.0.collection_time_point_relative\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "first_symptoms                  67\n",
       "Symptom onset                   58\n",
       "0 d                             19\n",
       "Symptom onset                   16\n",
       "Hospital Admission              14\n",
       "First negative COVID-19 test    10\n",
       "Name: sample.0.collection_time_point_reference, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"sample.0.collection_time_point_reference\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "assert not df[\"sample.0.collection_time_point_relative\"].isna().any()\n",
    "assert not df[\"sample.0.collection_time_point_reference\"].isna().any()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['repertoire_id', 'repertoire_name', 'repertoire_description', 'study.study_id', 'study.study_title', 'study.study_type.label', 'study.study_type.id', 'study.study_description', 'study.inclusion_exclusion_criteria', 'study.grants',\n",
       "       ...\n",
       "       'data_processing.0.software_versions', 'data_processing.0.paired_reads_assembly', 'data_processing.0.quality_thresholds', 'data_processing.0.primer_match_cutoffs', 'data_processing.0.collapsing_method', 'data_processing.0.data_processing_protocols', 'data_processing.0.data_processing_files.0', 'data_processing.0.germline_database', 'data_processing.0.analysis_provenance_id', 'study.keywords_study.1'], dtype='object', length=103)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "repertoire_id\n",
      "repertoire_name\n",
      "repertoire_description\n",
      "study.study_id\n",
      "study.study_title\n",
      "study.study_type.label\n",
      "study.study_type.id\n",
      "study.study_description\n",
      "study.inclusion_exclusion_criteria\n",
      "study.grants\n",
      "study.collected_by\n",
      "study.lab_name\n",
      "study.lab_address\n",
      "study.submitted_by\n",
      "study.pub_ids\n",
      "study.keywords_study.0\n",
      "subject.subject_id\n",
      "subject.synthetic\n",
      "subject.species.label\n",
      "subject.species.id\n",
      "subject.organism.label\n",
      "subject.organism.id\n",
      "subject.sex\n",
      "subject.age\n",
      "subject.age_min\n",
      "subject.age_max\n",
      "subject.age_unit.label\n",
      "subject.age_unit.id\n",
      "subject.age_event\n",
      "subject.ancestry_population\n",
      "subject.ethnicity\n",
      "subject.race\n",
      "subject.strain_name\n",
      "subject.linked_subjects\n",
      "subject.link_type\n",
      "subject.diagnosis.0.study_group_description\n",
      "subject.diagnosis.0.disease_diagnosis.label\n",
      "subject.diagnosis.0.disease_diagnosis.id\n",
      "subject.diagnosis.0.disease_length\n",
      "subject.diagnosis.0.disease_stage\n",
      "subject.diagnosis.0.prior_therapies\n",
      "subject.diagnosis.0.immunogen\n",
      "subject.diagnosis.0.intervention\n",
      "subject.diagnosis.0.medical_history\n",
      "sample.0.sample_processing_id\n",
      "sample.0.sample_id\n",
      "sample.0.sample_type\n",
      "sample.0.tissue.label\n",
      "sample.0.tissue.id\n",
      "sample.0.anatomic_site\n",
      "sample.0.disease_state_sample\n",
      "sample.0.collection_time_point_relative\n",
      "sample.0.collection_time_point_reference\n",
      "sample.0.biomaterial_provider\n",
      "sample.0.tissue_processing\n",
      "sample.0.cell_subset.label\n",
      "sample.0.cell_subset.id\n",
      "sample.0.cell_phenotype\n",
      "sample.0.cell_species.label\n",
      "sample.0.cell_species.id\n",
      "sample.0.single_cell\n",
      "sample.0.cell_number\n",
      "sample.0.cells_per_reaction\n",
      "sample.0.cell_storage\n",
      "sample.0.cell_quality\n",
      "sample.0.cell_isolation\n",
      "sample.0.cell_processing_protocol\n",
      "sample.0.template_class\n",
      "sample.0.template_quality\n",
      "sample.0.template_amount\n",
      "sample.0.library_generation_method\n",
      "sample.0.library_generation_protocol\n",
      "sample.0.library_generation_kit_version\n",
      "sample.0.pcr_target.0.pcr_target_locus\n",
      "sample.0.pcr_target.0.forward_pcr_primer_target_location\n",
      "sample.0.pcr_target.0.reverse_pcr_primer_target_location\n",
      "sample.0.complete_sequences\n",
      "sample.0.physical_linkage\n",
      "sample.0.sequencing_run_id\n",
      "sample.0.total_reads_passing_qc_filter\n",
      "sample.0.sequencing_platform\n",
      "sample.0.sequencing_facility\n",
      "sample.0.sequencing_run_date\n",
      "sample.0.sequencing_kit\n",
      "sample.0.sequencing_files.file_type\n",
      "sample.0.sequencing_files.filename\n",
      "sample.0.sequencing_files.read_direction\n",
      "sample.0.sequencing_files.read_length\n",
      "sample.0.sequencing_files.paired_filename\n",
      "sample.0.sequencing_files.paired_read_direction\n",
      "sample.0.sequencing_files.paired_read_length\n",
      "data_processing.0.data_processing_id\n",
      "data_processing.0.primary_annotation\n",
      "data_processing.0.software_versions\n",
      "data_processing.0.paired_reads_assembly\n",
      "data_processing.0.quality_thresholds\n",
      "data_processing.0.primer_match_cutoffs\n",
      "data_processing.0.collapsing_method\n",
      "data_processing.0.data_processing_protocols\n",
      "data_processing.0.data_processing_files.0\n",
      "data_processing.0.germline_database\n",
      "data_processing.0.analysis_provenance_id\n",
      "study.keywords_study.1\n"
     ]
    }
   ],
   "source": [
    "print(\"\\n\".join(df.columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study.study_id</th>\n",
       "      <th>study.study_title</th>\n",
       "      <th>study.study_type.label</th>\n",
       "      <th>study.study_type.id</th>\n",
       "      <th>study.study_description</th>\n",
       "      <th>study.inclusion_exclusion_criteria</th>\n",
       "      <th>study.grants</th>\n",
       "      <th>study.collected_by</th>\n",
       "      <th>study.lab_name</th>\n",
       "      <th>study.lab_address</th>\n",
       "      <th>study.submitted_by</th>\n",
       "      <th>study.pub_ids</th>\n",
       "      <th>study.keywords_study.0</th>\n",
       "      <th>study.keywords_study.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRJNA628125</td>\n",
       "      <td>B cell clonal expansion and convergent antibod...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>COVID-19 Study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...</td>\n",
       "      <td>S. Boyd, sboyd1@stanford.edu</td>\n",
       "      <td>Scott Boyd Lab, Department of Pathology</td>\n",
       "      <td>Stanford University</td>\n",
       "      <td>S. Nielsen, F. Yang, S. Boyd</td>\n",
       "      <td>DOI:10.21203/rs.3.rs-27220/v1</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PRJNA638224</td>\n",
       "      <td>Deep sequencing of B cell receptor repertoires...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>To drive a deeper understanding of the nature ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Engineering and Physical Sciences Research Cou...</td>\n",
       "      <td>jake@alchemab.com,  jane@alchemab.com</td>\n",
       "      <td>Alchemab Therapeutics Ltd</td>\n",
       "      <td>55-56 Russell Square, London, WC1B 4HP, UK</td>\n",
       "      <td>Jacob Galson, jake@alchemab.com</td>\n",
       "      <td>DOI:10.1101/2020.05.20.106294</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>IR-Binder-000001</td>\n",
       "      <td>Next Generation Sequencing of T and B cell rec...</td>\n",
       "      <td>Study</td>\n",
       "      <td>NCIT:C63536</td>\n",
       "      <td>Immunological profiling of Covid-19 patients w...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mascha.Binder@uk-halle.de, Donjete.Simnica@uk-...</td>\n",
       "      <td>Department of Internal Medicine IV, Oncology/H...</td>\n",
       "      <td>Martin-Luther-University Halle-Wittenberg</td>\n",
       "      <td>Edith.Willscher@uk-halle.de</td>\n",
       "      <td>DOI: 10.1016/j.immuni.2020.06.024</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>Stereotypic Neutralizing VH Clonotypes Against...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>In response to SARS-CoV-2 infection, most huma...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The National Research Foundation of Korea [NRF...</td>\n",
       "      <td>Sunghoon Kwon skwon@snu.ac.kr; Junho Chung  jj...</td>\n",
       "      <td>Department of Biochemistry and Molecular Biology</td>\n",
       "      <td>Seoul National University College of Medicine</td>\n",
       "      <td>Sang Il Kim, Jinsung Noh, Sunghoon Kwon, Junho...</td>\n",
       "      <td>DOI: 10.1101/2020.06.26.174557</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>PRJNA630455</td>\n",
       "      <td>Comprehensive mapping of immune perturbations ...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>Identified extensive induction and activation ...</td>\n",
       "      <td>Two severe COVID-19+ individuals were excluded...</td>\n",
       "      <td>The University of Pennsylvania Institute for I...</td>\n",
       "      <td>Leticia Kuri-Cervantes, M. Betina Pampena, E. ...</td>\n",
       "      <td>Department of Microbiology and Institute for I...</td>\n",
       "      <td>Perelman School of Medicine, University of Pen...</td>\n",
       "      <td>Michael R. Betts (betts@pennmedicine.upenn.edu)</td>\n",
       "      <td>PMID: 32669287</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>PRJCA002413</td>\n",
       "      <td>Immune cell profiling of COVID-19 patients in ...</td>\n",
       "      <td>Case-Control Study</td>\n",
       "      <td>NCIT:C15197</td>\n",
       "      <td>An inflammatory immune signature in the early ...</td>\n",
       "      <td>Include: COVID</td>\n",
       "      <td>National Natural Science Foundation of China, ...</td>\n",
       "      <td>xiaochuanle@126.com, cw0226@foxmail.com, hywan...</td>\n",
       "      <td>National Center for Liver Cancer</td>\n",
       "      <td>Second Military Medical University, 200438 Sha...</td>\n",
       "      <td>Chuanle Xiao, xiaochuanle@126.com</td>\n",
       "      <td>PMID:32377375</td>\n",
       "      <td>contains_ig</td>\n",
       "      <td>contains_paired_chain</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       study.study_id                                  study.study_title study.study_type.label study.study_type.id                            study.study_description                 study.inclusion_exclusion_criteria                                       study.grants                                 study.collected_by                                     study.lab_name                                  study.lab_address                                 study.submitted_by                      study.pub_ids study.keywords_study.0 study.keywords_study.1\n",
       "0         PRJNA628125  B cell clonal expansion and convergent antibod...     Case-Control Study         NCIT:C15197                                     COVID-19 Study                                                NaN  NIH/NIAID T32AI007502-23; NIH/NHLBI K23HL12566...                       S. Boyd, sboyd1@stanford.edu            Scott Boyd Lab, Department of Pathology                                Stanford University                       S. Nielsen, F. Yang, S. Boyd      DOI:10.21203/rs.3.rs-27220/v1            contains_ig                    NaN\n",
       "14        PRJNA638224  Deep sequencing of B cell receptor repertoires...     Case-Control Study         NCIT:C15197  To drive a deeper understanding of the nature ...                                                NaN  Engineering and Physical Sciences Research Cou...              jake@alchemab.com,  jane@alchemab.com                          Alchemab Therapeutics Ltd         55-56 Russell Square, London, WC1B 4HP, UK                    Jacob Galson, jake@alchemab.com      DOI:10.1101/2020.05.20.106294            contains_ig                    NaN\n",
       "33   IR-Binder-000001  Next Generation Sequencing of T and B cell rec...                  Study         NCIT:C63536  Immunological profiling of Covid-19 patients w...                                                NaN                                                NaN  Mascha.Binder@uk-halle.de, Donjete.Simnica@uk-...  Department of Internal Medicine IV, Oncology/H...          Martin-Luther-University Halle-Wittenberg                        Edith.Willscher@uk-halle.de  DOI: 10.1016/j.immuni.2020.06.024            contains_ig                    NaN\n",
       "100       PRJNA648677  Stereotypic Neutralizing VH Clonotypes Against...     Case-Control Study         NCIT:C15197  In response to SARS-CoV-2 infection, most huma...                                                NaN  The National Research Foundation of Korea [NRF...  Sunghoon Kwon skwon@snu.ac.kr; Junho Chung  jj...   Department of Biochemistry and Molecular Biology      Seoul National University College of Medicine  Sang Il Kim, Jinsung Noh, Sunghoon Kwon, Junho...     DOI: 10.1101/2020.06.26.174557            contains_ig                    NaN\n",
       "116       PRJNA630455  Comprehensive mapping of immune perturbations ...     Case-Control Study         NCIT:C15197  Identified extensive induction and activation ...  Two severe COVID-19+ individuals were excluded...  The University of Pennsylvania Institute for I...  Leticia Kuri-Cervantes, M. Betina Pampena, E. ...  Department of Microbiology and Institute for I...  Perelman School of Medicine, University of Pen...    Michael R. Betts (betts@pennmedicine.upenn.edu)                     PMID: 32669287            contains_ig                    NaN\n",
       "174       PRJCA002413  Immune cell profiling of COVID-19 patients in ...     Case-Control Study         NCIT:C15197  An inflammatory immune signature in the early ...                                     Include: COVID  National Natural Science Foundation of China, ...  xiaochuanle@126.com, cw0226@foxmail.com, hywan...                   National Center for Liver Cancer  Second Military Medical University, 200438 Sha...                  Chuanle Xiao, xiaochuanle@126.com                     PMID:32377375            contains_ig  contains_paired_chain"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.columns[df.columns.str.startswith(\"study.\")]].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'PRJNA648677': 'Kim'}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_names = {\"PRJNA648677\": \"Kim\"}  # could add more here\n",
    "study_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study.study_id</th>\n",
       "      <th>subject.subject_id</th>\n",
       "      <th>subject.synthetic</th>\n",
       "      <th>subject.species.label</th>\n",
       "      <th>subject.species.id</th>\n",
       "      <th>subject.sex</th>\n",
       "      <th>subject.age_min</th>\n",
       "      <th>subject.age_max</th>\n",
       "      <th>subject.age_unit.label</th>\n",
       "      <th>subject.age_unit.id</th>\n",
       "      <th>subject.age_event</th>\n",
       "      <th>subject.race</th>\n",
       "      <th>subject.diagnosis.0.study_group_description</th>\n",
       "      <th>subject.diagnosis.0.disease_diagnosis.label</th>\n",
       "      <th>subject.diagnosis.0.disease_diagnosis.id</th>\n",
       "      <th>subject.diagnosis.0.disease_length</th>\n",
       "      <th>subject.diagnosis.0.disease_stage</th>\n",
       "      <th>subject.diagnosis.0.prior_therapies</th>\n",
       "      <th>subject.diagnosis.0.medical_history</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>11 days since symptom onset</td>\n",
       "      <td>Extensive Pneumonic infiltrates</td>\n",
       "      <td>Oxygen therapy, Lopinavir/ritonavir, Levofloxacin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>17 days since symptom onset</td>\n",
       "      <td>Extensive Pneumonic infiltrates</td>\n",
       "      <td>Oxygen therapy, Lopinavir/ritonavir, Levofloxacin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>45 days since symptom onset</td>\n",
       "      <td>Extensive Pneumonic infiltrates</td>\n",
       "      <td>Oxygen therapy, Lopinavir/ritonavir, Levofloxacin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>10 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diabetes mellitus, dyslipidemia, hypertension</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>19 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diabetes mellitus, dyslipidemia, hypertension</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>female</td>\n",
       "      <td>53.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>6 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>female</td>\n",
       "      <td>53.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>15 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>24.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>6 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>24.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>28 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>23 days since symptom onset</td>\n",
       "      <td>Extensive Pneumonic infiltrates</td>\n",
       "      <td>Lopinavir/ritonavir</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>44 days since symptom onset</td>\n",
       "      <td>Extensive Pneumonic infiltrates</td>\n",
       "      <td>Lopinavir/ritonavir</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>99 days since symptom onset</td>\n",
       "      <td>Extensive Pneumonic infiltrates</td>\n",
       "      <td>Lopinavir/ritonavir</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>female</td>\n",
       "      <td>40.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>14 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>Lopinavir/ritonavir</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>female</td>\n",
       "      <td>40.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>36 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>Lopinavir/ritonavir</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>female</td>\n",
       "      <td>59.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>9 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diabetes mellitus, dyslipidemia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>False</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>female</td>\n",
       "      <td>59.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>year</td>\n",
       "      <td>UO:0000036</td>\n",
       "      <td>Sample Collection</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>DOID:0080600</td>\n",
       "      <td>22 days since symptom onset</td>\n",
       "      <td>Limited Pneumonic infiltrates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>diabetes mellitus, dyslipidemia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id study.study_id subject.subject_id  subject.synthetic subject.species.label subject.species.id subject.sex  subject.age_min  subject.age_max subject.age_unit.label subject.age_unit.id  subject.age_event subject.race subject.diagnosis.0.study_group_description subject.diagnosis.0.disease_diagnosis.label subject.diagnosis.0.disease_diagnosis.id subject.diagnosis.0.disease_length subject.diagnosis.0.disease_stage                subject.diagnosis.0.prior_therapies            subject.diagnosis.0.medical_history\n",
       "100  5f21e814e1adeb2edc12613c    PRJNA648677                  A              False          Homo sapiens     NCBITAXON:9606        male             55.0             55.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600        11 days since symptom onset   Extensive Pneumonic infiltrates  Oxygen therapy, Lopinavir/ritonavir, Levofloxacin                                            NaN\n",
       "101  5f21e814e1adeb2edc12613d    PRJNA648677                  A              False          Homo sapiens     NCBITAXON:9606        male             55.0             55.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600        17 days since symptom onset   Extensive Pneumonic infiltrates  Oxygen therapy, Lopinavir/ritonavir, Levofloxacin                                            NaN\n",
       "102  5f21e815e1adeb2edc12613e    PRJNA648677                  A              False          Homo sapiens     NCBITAXON:9606        male             55.0             55.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600        45 days since symptom onset   Extensive Pneumonic infiltrates  Oxygen therapy, Lopinavir/ritonavir, Levofloxacin                                            NaN\n",
       "103  5f21e815e1adeb2edc12613f    PRJNA648677                  B              False          Homo sapiens     NCBITAXON:9606        male             55.0             55.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600        10 days since symptom onset     Limited Pneumonic infiltrates                                                NaN  diabetes mellitus, dyslipidemia, hypertension\n",
       "104  5f21e815e1adeb2edc126140    PRJNA648677                  B              False          Homo sapiens     NCBITAXON:9606        male             55.0             55.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600        19 days since symptom onset     Limited Pneumonic infiltrates                                                NaN  diabetes mellitus, dyslipidemia, hypertension\n",
       "105  5f21e816e1adeb2edc126141    PRJNA648677                  C              False          Homo sapiens     NCBITAXON:9606      female             53.0             53.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600         6 days since symptom onset     Limited Pneumonic infiltrates                                                NaN                                            NaN\n",
       "106  5f21e816e1adeb2edc126142    PRJNA648677                  C              False          Homo sapiens     NCBITAXON:9606      female             53.0             53.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600        15 days since symptom onset     Limited Pneumonic infiltrates                                                NaN                                            NaN\n",
       "107  5f21e816e1adeb2edc126143    PRJNA648677                  D              False          Homo sapiens     NCBITAXON:9606        male             24.0             24.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600         6 days since symptom onset     Limited Pneumonic infiltrates                                                NaN                                            NaN\n",
       "108  5f21e817e1adeb2edc126144    PRJNA648677                  D              False          Homo sapiens     NCBITAXON:9606        male             24.0             24.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600        28 days since symptom onset     Limited Pneumonic infiltrates                                                NaN                                            NaN\n",
       "109  5f21e817e1adeb2edc126145    PRJNA648677                  E              False          Homo sapiens     NCBITAXON:9606        male             48.0             48.0                   year          UO:0000036  Sample Collection      Chinese                                        Case                                    COVID-19                             DOID:0080600        23 days since symptom onset   Extensive Pneumonic infiltrates                                Lopinavir/ritonavir                                            NaN\n",
       "110  5f21e817e1adeb2edc126146    PRJNA648677                  E              False          Homo sapiens     NCBITAXON:9606        male             48.0             48.0                   year          UO:0000036  Sample Collection      Chinese                                        Case                                    COVID-19                             DOID:0080600        44 days since symptom onset   Extensive Pneumonic infiltrates                                Lopinavir/ritonavir                                            NaN\n",
       "111  5f21e818e1adeb2edc126147    PRJNA648677                  E              False          Homo sapiens     NCBITAXON:9606        male             48.0             48.0                   year          UO:0000036  Sample Collection      Chinese                                        Case                                    COVID-19                             DOID:0080600        99 days since symptom onset   Extensive Pneumonic infiltrates                                Lopinavir/ritonavir                                            NaN\n",
       "112  5f21e818e1adeb2edc126148    PRJNA648677                  F              False          Homo sapiens     NCBITAXON:9606      female             40.0             40.0                   year          UO:0000036  Sample Collection      Chinese                                        Case                                    COVID-19                             DOID:0080600        14 days since symptom onset     Limited Pneumonic infiltrates                                Lopinavir/ritonavir                                            NaN\n",
       "113  5f21e818e1adeb2edc126149    PRJNA648677                  F              False          Homo sapiens     NCBITAXON:9606      female             40.0             40.0                   year          UO:0000036  Sample Collection      Chinese                                        Case                                    COVID-19                             DOID:0080600        36 days since symptom onset     Limited Pneumonic infiltrates                                Lopinavir/ritonavir                                            NaN\n",
       "114  5f21e818e1adeb2edc12614a    PRJNA648677                  G              False          Homo sapiens     NCBITAXON:9606      female             59.0             59.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600         9 days since symptom onset     Limited Pneumonic infiltrates                                                NaN                diabetes mellitus, dyslipidemia\n",
       "115  5f21e819e1adeb2edc12614b    PRJNA648677                  G              False          Homo sapiens     NCBITAXON:9606      female             59.0             59.0                   year          UO:0000036  Sample Collection       Korean                                        Case                                    COVID-19                             DOID:0080600        22 days since symptom onset     Limited Pneumonic infiltrates                                                NaN                diabetes mellitus, dyslipidemia"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for study_id in study_names.keys():\n",
    "    display(\n",
    "        df[df[\"study.study_id\"] == study_id][\n",
    "            [\"repertoire_id\", \"study.study_id\"]\n",
    "            + df.columns[df.columns.str.startswith(\"subject.\")].tolist()\n",
    "        ]\n",
    "        .dropna(how=\"all\", axis=1)\n",
    "        .drop_duplicates()\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study.study_id</th>\n",
       "      <th>subject.subject_id</th>\n",
       "      <th>sample.0.sample_processing_id</th>\n",
       "      <th>sample.0.sample_id</th>\n",
       "      <th>sample.0.sample_type</th>\n",
       "      <th>sample.0.tissue.label</th>\n",
       "      <th>sample.0.tissue.id</th>\n",
       "      <th>sample.0.collection_time_point_relative</th>\n",
       "      <th>sample.0.collection_time_point_reference</th>\n",
       "      <th>sample.0.biomaterial_provider</th>\n",
       "      <th>sample.0.tissue_processing</th>\n",
       "      <th>sample.0.cell_species.label</th>\n",
       "      <th>sample.0.cell_species.id</th>\n",
       "      <th>sample.0.single_cell</th>\n",
       "      <th>sample.0.cell_storage</th>\n",
       "      <th>sample.0.cell_processing_protocol</th>\n",
       "      <th>sample.0.template_class</th>\n",
       "      <th>sample.0.template_amount</th>\n",
       "      <th>sample.0.library_generation_method</th>\n",
       "      <th>sample.0.library_generation_protocol</th>\n",
       "      <th>sample.0.pcr_target.0.pcr_target_locus</th>\n",
       "      <th>sample.0.pcr_target.0.forward_pcr_primer_target_location</th>\n",
       "      <th>sample.0.pcr_target.0.reverse_pcr_primer_target_location</th>\n",
       "      <th>sample.0.complete_sequences</th>\n",
       "      <th>sample.0.physical_linkage</th>\n",
       "      <th>sample.0.total_reads_passing_qc_filter</th>\n",
       "      <th>sample.0.sequencing_platform</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>A_d11</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d11</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>294633.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>A_d17</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d17</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>240012.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>A_d45</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d45</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>215355.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>B_d10</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d10</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>97819.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>B_d19</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d19</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>250000.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>C_d6</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d6</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>246327.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>C_d15</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d15</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>250000.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>D_d6</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d6</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>246333.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>D_d28</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d28</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>142798.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>E_d23</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d23</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>205326.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>E_d44</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d44</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>234353.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>E_d99</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d99</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>182221.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>F_d14</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d14</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>179069.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>F_d36</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d36</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>179398.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>G_d9</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d9</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>229788.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>G_d22</td>\n",
       "      <td>chronological blood samples were drawn</td>\n",
       "      <td>blood</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>d22</td>\n",
       "      <td>Symptom onset</td>\n",
       "      <td>Seoul National University Hospital</td>\n",
       "      <td>PBMCs and plasma were isolated using Lymphopre...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NCBITAXON:9606</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>The PBMCs were subjected to total RNA isolatio...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>500 ng</td>\n",
       "      <td>RT(specific+UMI)+PCR</td>\n",
       "      <td>Sampled 5ug of mRNA was used to synthesize cDN...</td>\n",
       "      <td>IGH</td>\n",
       "      <td>IGHV</td>\n",
       "      <td>IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...</td>\n",
       "      <td>partial</td>\n",
       "      <td>none</td>\n",
       "      <td>250000.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id study.study_id subject.subject_id sample.0.sample_processing_id sample.0.sample_id                    sample.0.sample_type sample.0.tissue.label sample.0.tissue.id sample.0.collection_time_point_relative sample.0.collection_time_point_reference       sample.0.biomaterial_provider                         sample.0.tissue_processing sample.0.cell_species.label sample.0.cell_species.id  sample.0.single_cell sample.0.cell_storage                  sample.0.cell_processing_protocol sample.0.template_class sample.0.template_amount sample.0.library_generation_method               sample.0.library_generation_protocol sample.0.pcr_target.0.pcr_target_locus sample.0.pcr_target.0.forward_pcr_primer_target_location sample.0.pcr_target.0.reverse_pcr_primer_target_location sample.0.complete_sequences sample.0.physical_linkage  sample.0.total_reads_passing_qc_filter sample.0.sequencing_platform\n",
       "100  5f21e814e1adeb2edc12613c    PRJNA648677                  A      5f21e814e1adeb2edc12613c              A_d11  chronological blood samples were drawn                 blood     UBERON:0000178                                     d11                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                294633.0               Illumina MiSeq\n",
       "101  5f21e814e1adeb2edc12613d    PRJNA648677                  A      5f21e814e1adeb2edc12613d              A_d17  chronological blood samples were drawn                 blood     UBERON:0000178                                     d17                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                240012.0               Illumina MiSeq\n",
       "102  5f21e815e1adeb2edc12613e    PRJNA648677                  A      5f21e815e1adeb2edc12613e              A_d45  chronological blood samples were drawn                 blood     UBERON:0000178                                     d45                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                215355.0               Illumina MiSeq\n",
       "103  5f21e815e1adeb2edc12613f    PRJNA648677                  B      5f21e815e1adeb2edc12613f              B_d10  chronological blood samples were drawn                 blood     UBERON:0000178                                     d10                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                 97819.0               Illumina MiSeq\n",
       "104  5f21e815e1adeb2edc126140    PRJNA648677                  B      5f21e815e1adeb2edc126140              B_d19  chronological blood samples were drawn                 blood     UBERON:0000178                                     d19                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                250000.0               Illumina MiSeq\n",
       "105  5f21e816e1adeb2edc126141    PRJNA648677                  C      5f21e816e1adeb2edc126141               C_d6  chronological blood samples were drawn                 blood     UBERON:0000178                                      d6                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                246327.0               Illumina MiSeq\n",
       "106  5f21e816e1adeb2edc126142    PRJNA648677                  C      5f21e816e1adeb2edc126142              C_d15  chronological blood samples were drawn                 blood     UBERON:0000178                                     d15                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                250000.0               Illumina MiSeq\n",
       "107  5f21e816e1adeb2edc126143    PRJNA648677                  D      5f21e816e1adeb2edc126143               D_d6  chronological blood samples were drawn                 blood     UBERON:0000178                                      d6                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                246333.0               Illumina MiSeq\n",
       "108  5f21e817e1adeb2edc126144    PRJNA648677                  D      5f21e817e1adeb2edc126144              D_d28  chronological blood samples were drawn                 blood     UBERON:0000178                                     d28                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                142798.0               Illumina MiSeq\n",
       "109  5f21e817e1adeb2edc126145    PRJNA648677                  E      5f21e817e1adeb2edc126145              E_d23  chronological blood samples were drawn                 blood     UBERON:0000178                                     d23                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                205326.0               Illumina MiSeq\n",
       "110  5f21e817e1adeb2edc126146    PRJNA648677                  E      5f21e817e1adeb2edc126146              E_d44  chronological blood samples were drawn                 blood     UBERON:0000178                                     d44                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                234353.0               Illumina MiSeq\n",
       "111  5f21e818e1adeb2edc126147    PRJNA648677                  E      5f21e818e1adeb2edc126147              E_d99  chronological blood samples were drawn                 blood     UBERON:0000178                                     d99                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                182221.0               Illumina MiSeq\n",
       "112  5f21e818e1adeb2edc126148    PRJNA648677                  F      5f21e818e1adeb2edc126148              F_d14  chronological blood samples were drawn                 blood     UBERON:0000178                                     d14                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                179069.0               Illumina MiSeq\n",
       "113  5f21e818e1adeb2edc126149    PRJNA648677                  F      5f21e818e1adeb2edc126149              F_d36  chronological blood samples were drawn                 blood     UBERON:0000178                                     d36                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                179398.0               Illumina MiSeq\n",
       "114  5f21e818e1adeb2edc12614a    PRJNA648677                  G      5f21e818e1adeb2edc12614a               G_d9  chronological blood samples were drawn                 blood     UBERON:0000178                                      d9                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                229788.0               Illumina MiSeq\n",
       "115  5f21e819e1adeb2edc12614b    PRJNA648677                  G      5f21e819e1adeb2edc12614b              G_d22  chronological blood samples were drawn                 blood     UBERON:0000178                                     d22                            Symptom onset  Seoul National University Hospital  PBMCs and plasma were isolated using Lymphopre...                Homo sapiens           NCBITAXON:9606                 False                 False  The PBMCs were subjected to total RNA isolatio...                     RNA                   500 ng               RT(specific+UMI)+PCR  Sampled 5ug of mRNA was used to synthesize cDN...                                    IGH                                               IGHV        IGHC1 of each isotype (IgM, IgD, IgG, IgA, and...                           partial                      none                                250000.0               Illumina MiSeq"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for study_id in study_names.keys():\n",
    "    display(\n",
    "        df[df[\"study.study_id\"] == study_id][\n",
    "            [\"repertoire_id\", \"study.study_id\", \"subject.subject_id\"]\n",
    "            + df.columns[df.columns.str.startswith(\"sample.\")].tolist()\n",
    "        ]\n",
    "        .dropna(how=\"all\", axis=1)\n",
    "        .drop_duplicates()\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study.study_id</th>\n",
       "      <th>subject.subject_id</th>\n",
       "      <th>subject.sex</th>\n",
       "      <th>subject.age_min</th>\n",
       "      <th>subject.race</th>\n",
       "      <th>subject.diagnosis.0.study_group_description</th>\n",
       "      <th>subject.diagnosis.0.disease_diagnosis.label</th>\n",
       "      <th>sample.0.sample_id</th>\n",
       "      <th>sample.0.collection_time_point_relative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>A_d11</td>\n",
       "      <td>d11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>A_d17</td>\n",
       "      <td>d17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>A_d45</td>\n",
       "      <td>d45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>B_d10</td>\n",
       "      <td>d10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>B_d19</td>\n",
       "      <td>d19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>female</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>C_d6</td>\n",
       "      <td>d6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>female</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>C_d15</td>\n",
       "      <td>d15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>male</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>D_d6</td>\n",
       "      <td>d6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>male</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>D_d28</td>\n",
       "      <td>d28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>E_d23</td>\n",
       "      <td>d23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>E_d44</td>\n",
       "      <td>d44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>E_d99</td>\n",
       "      <td>d99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>female</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>F_d14</td>\n",
       "      <td>d14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>female</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>F_d36</td>\n",
       "      <td>d36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>female</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>G_d9</td>\n",
       "      <td>d9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>female</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>G_d22</td>\n",
       "      <td>d22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id study.study_id subject.subject_id subject.sex  subject.age_min subject.race subject.diagnosis.0.study_group_description subject.diagnosis.0.disease_diagnosis.label sample.0.sample_id sample.0.collection_time_point_relative\n",
       "100  5f21e814e1adeb2edc12613c    PRJNA648677                  A        male             55.0       Korean                                        Case                                    COVID-19              A_d11                                     d11\n",
       "101  5f21e814e1adeb2edc12613d    PRJNA648677                  A        male             55.0       Korean                                        Case                                    COVID-19              A_d17                                     d17\n",
       "102  5f21e815e1adeb2edc12613e    PRJNA648677                  A        male             55.0       Korean                                        Case                                    COVID-19              A_d45                                     d45\n",
       "103  5f21e815e1adeb2edc12613f    PRJNA648677                  B        male             55.0       Korean                                        Case                                    COVID-19              B_d10                                     d10\n",
       "104  5f21e815e1adeb2edc126140    PRJNA648677                  B        male             55.0       Korean                                        Case                                    COVID-19              B_d19                                     d19\n",
       "105  5f21e816e1adeb2edc126141    PRJNA648677                  C      female             53.0       Korean                                        Case                                    COVID-19               C_d6                                      d6\n",
       "106  5f21e816e1adeb2edc126142    PRJNA648677                  C      female             53.0       Korean                                        Case                                    COVID-19              C_d15                                     d15\n",
       "107  5f21e816e1adeb2edc126143    PRJNA648677                  D        male             24.0       Korean                                        Case                                    COVID-19               D_d6                                      d6\n",
       "108  5f21e817e1adeb2edc126144    PRJNA648677                  D        male             24.0       Korean                                        Case                                    COVID-19              D_d28                                     d28\n",
       "109  5f21e817e1adeb2edc126145    PRJNA648677                  E        male             48.0      Chinese                                        Case                                    COVID-19              E_d23                                     d23\n",
       "110  5f21e817e1adeb2edc126146    PRJNA648677                  E        male             48.0      Chinese                                        Case                                    COVID-19              E_d44                                     d44\n",
       "111  5f21e818e1adeb2edc126147    PRJNA648677                  E        male             48.0      Chinese                                        Case                                    COVID-19              E_d99                                     d99\n",
       "112  5f21e818e1adeb2edc126148    PRJNA648677                  F      female             40.0      Chinese                                        Case                                    COVID-19              F_d14                                     d14\n",
       "113  5f21e818e1adeb2edc126149    PRJNA648677                  F      female             40.0      Chinese                                        Case                                    COVID-19              F_d36                                     d36\n",
       "114  5f21e818e1adeb2edc12614a    PRJNA648677                  G      female             59.0       Korean                                        Case                                    COVID-19               G_d9                                      d9\n",
       "115  5f21e819e1adeb2edc12614b    PRJNA648677                  G      female             59.0       Korean                                        Case                                    COVID-19              G_d22                                     d22"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for study_id in study_names.keys():\n",
    "    display(\n",
    "        df[df[\"study.study_id\"] == study_id][\n",
    "            [\"repertoire_id\", \"study.study_id\", \"subject.subject_id\"]\n",
    "            + [\n",
    "                \"subject.sex\",\n",
    "                \"subject.age_min\",\n",
    "                \"subject.race\",\n",
    "                \"subject.diagnosis.0.study_group_description\",\n",
    "                \"subject.diagnosis.0.disease_diagnosis.label\",\n",
    "            ]\n",
    "            + [\"sample.0.sample_id\", \"sample.0.collection_time_point_relative\"]\n",
    "        ]\n",
    "        .dropna(how=\"all\", axis=1)\n",
    "        .drop_duplicates()\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## create patient IDs, and extract some patient-level metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study.study_id</th>\n",
       "      <th>subject.subject_id</th>\n",
       "      <th>subject.sex</th>\n",
       "      <th>subject.age_min</th>\n",
       "      <th>subject.race</th>\n",
       "      <th>subject.diagnosis.0.study_group_description</th>\n",
       "      <th>subject.diagnosis.0.disease_diagnosis.label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>male</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>female</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>female</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>male</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>male</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>female</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>female</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>female</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>female</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id study.study_id subject.subject_id subject.sex  subject.age_min subject.race subject.diagnosis.0.study_group_description subject.diagnosis.0.disease_diagnosis.label\n",
       "100  5f21e814e1adeb2edc12613c    PRJNA648677                  A        male             55.0       Korean                                        Case                                    COVID-19\n",
       "101  5f21e814e1adeb2edc12613d    PRJNA648677                  A        male             55.0       Korean                                        Case                                    COVID-19\n",
       "102  5f21e815e1adeb2edc12613e    PRJNA648677                  A        male             55.0       Korean                                        Case                                    COVID-19\n",
       "103  5f21e815e1adeb2edc12613f    PRJNA648677                  B        male             55.0       Korean                                        Case                                    COVID-19\n",
       "104  5f21e815e1adeb2edc126140    PRJNA648677                  B        male             55.0       Korean                                        Case                                    COVID-19\n",
       "105  5f21e816e1adeb2edc126141    PRJNA648677                  C      female             53.0       Korean                                        Case                                    COVID-19\n",
       "106  5f21e816e1adeb2edc126142    PRJNA648677                  C      female             53.0       Korean                                        Case                                    COVID-19\n",
       "107  5f21e816e1adeb2edc126143    PRJNA648677                  D        male             24.0       Korean                                        Case                                    COVID-19\n",
       "108  5f21e817e1adeb2edc126144    PRJNA648677                  D        male             24.0       Korean                                        Case                                    COVID-19\n",
       "109  5f21e817e1adeb2edc126145    PRJNA648677                  E        male             48.0      Chinese                                        Case                                    COVID-19\n",
       "110  5f21e817e1adeb2edc126146    PRJNA648677                  E        male             48.0      Chinese                                        Case                                    COVID-19\n",
       "111  5f21e818e1adeb2edc126147    PRJNA648677                  E        male             48.0      Chinese                                        Case                                    COVID-19\n",
       "112  5f21e818e1adeb2edc126148    PRJNA648677                  F      female             40.0      Chinese                                        Case                                    COVID-19\n",
       "113  5f21e818e1adeb2edc126149    PRJNA648677                  F      female             40.0      Chinese                                        Case                                    COVID-19\n",
       "114  5f21e818e1adeb2edc12614a    PRJNA648677                  G      female             59.0       Korean                                        Case                                    COVID-19\n",
       "115  5f21e819e1adeb2edc12614b    PRJNA648677                  G      female             59.0       Korean                                        Case                                    COVID-19"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# find the right columns...\n",
    "for study_id in study_names.keys():\n",
    "    display(\n",
    "        df[df[\"study.study_id\"] == study_id][\n",
    "            [\"repertoire_id\", \"study.study_id\", \"subject.subject_id\"]\n",
    "            + [\n",
    "                \"subject.sex\",\n",
    "                \"subject.age_min\",\n",
    "                \"subject.race\",\n",
    "                \"subject.diagnosis.0.study_group_description\",\n",
    "                \"subject.diagnosis.0.disease_diagnosis.label\",\n",
    "            ]\n",
    "        ]\n",
    "        .dropna(how=\"all\", axis=1)\n",
    "        .drop_duplicates()\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study_id</th>\n",
       "      <th>patient_id_within_study</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>disease_subtype</th>\n",
       "      <th>disease</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>specimen_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d11</td>\n",
       "      <td>A_d11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d17</td>\n",
       "      <td>A_d17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d45</td>\n",
       "      <td>A_d45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d10</td>\n",
       "      <td>B_d10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d19</td>\n",
       "      <td>B_d19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d6</td>\n",
       "      <td>C_d6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d15</td>\n",
       "      <td>C_d15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d6</td>\n",
       "      <td>D_d6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d28</td>\n",
       "      <td>D_d28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d23</td>\n",
       "      <td>E_d23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d44</td>\n",
       "      <td>E_d44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d99</td>\n",
       "      <td>E_d99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d14</td>\n",
       "      <td>F_d14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d36</td>\n",
       "      <td>F_d36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d9</td>\n",
       "      <td>G_d9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d22</td>\n",
       "      <td>G_d22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id     study_id patient_id_within_study sex   age ethnicity disease_subtype   disease timepoint specimen_label\n",
       "100  5f21e814e1adeb2edc12613c  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19       d11          A_d11\n",
       "101  5f21e814e1adeb2edc12613d  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19       d17          A_d17\n",
       "102  5f21e815e1adeb2edc12613e  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19       d45          A_d45\n",
       "103  5f21e815e1adeb2edc12613f  PRJNA648677                       B   M  55.0    Korean            Case  COVID-19       d10          B_d10\n",
       "104  5f21e815e1adeb2edc126140  PRJNA648677                       B   M  55.0    Korean            Case  COVID-19       d19          B_d19\n",
       "105  5f21e816e1adeb2edc126141  PRJNA648677                       C   F  53.0    Korean            Case  COVID-19        d6           C_d6\n",
       "106  5f21e816e1adeb2edc126142  PRJNA648677                       C   F  53.0    Korean            Case  COVID-19       d15          C_d15\n",
       "107  5f21e816e1adeb2edc126143  PRJNA648677                       D   M  24.0    Korean            Case  COVID-19        d6           D_d6\n",
       "108  5f21e817e1adeb2edc126144  PRJNA648677                       D   M  24.0    Korean            Case  COVID-19       d28          D_d28\n",
       "109  5f21e817e1adeb2edc126145  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19       d23          E_d23\n",
       "110  5f21e817e1adeb2edc126146  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19       d44          E_d44\n",
       "111  5f21e818e1adeb2edc126147  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19       d99          E_d99\n",
       "112  5f21e818e1adeb2edc126148  PRJNA648677                       F   F  40.0   Chinese            Case  COVID-19       d14          F_d14\n",
       "113  5f21e818e1adeb2edc126149  PRJNA648677                       F   F  40.0   Chinese            Case  COVID-19       d36          F_d36\n",
       "114  5f21e818e1adeb2edc12614a  PRJNA648677                       G   F  59.0    Korean            Case  COVID-19        d9           G_d9\n",
       "115  5f21e819e1adeb2edc12614b  PRJNA648677                       G   F  59.0    Korean            Case  COVID-19       d22          G_d22"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df = (\n",
    "    df[df[\"study.study_id\"].isin(study_names.keys())][\n",
    "        [\n",
    "            \"repertoire_id\",  # the internal repertoire ID\n",
    "            \"study.study_id\",\n",
    "            \"subject.subject_id\",\n",
    "            \"subject.sex\",\n",
    "            \"subject.age_min\",\n",
    "            \"subject.race\",\n",
    "            \"subject.diagnosis.0.study_group_description\",\n",
    "            \"subject.diagnosis.0.disease_diagnosis.label\",\n",
    "            \"sample.0.collection_time_point_relative\",\n",
    "            \"sample.0.sample_id\",\n",
    "        ]\n",
    "    ]\n",
    "    .dropna(how=\"all\", axis=1)\n",
    "    .drop_duplicates()\n",
    "    .rename(\n",
    "        columns={\n",
    "            \"subject.subject_id\": \"patient_id_within_study\",\n",
    "            \"study.study_id\": \"study_id\",\n",
    "            \"subject.sex\": \"sex\",\n",
    "            \"subject.age_min\": \"age\",\n",
    "            \"subject.race\": \"ethnicity\",\n",
    "            \"subject.diagnosis.0.study_group_description\": \"disease_subtype\",\n",
    "            \"subject.diagnosis.0.disease_diagnosis.label\": \"disease\",\n",
    "            \"sample.0.collection_time_point_relative\": \"timepoint\",\n",
    "            \"sample.0.sample_id\": \"specimen_label\",\n",
    "        }\n",
    "    )\n",
    ")\n",
    "specimens_df[\"sex\"] = specimens_df[\"sex\"].replace({\"male\": \"M\", \"female\": \"F\"})\n",
    "specimens_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Korean     11\n",
       "Chinese     5\n",
       "Name: ethnicity, dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df[\"ethnicity\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False    16\n",
       "Name: ethnicity, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df[\"ethnicity\"].isna().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Asian    16\n",
       "Name: ethnicity_condensed, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create ethnicity_condensed\n",
    "specimens_df[\"ethnicity_condensed\"] = specimens_df[\"ethnicity\"].replace(\n",
    "    {\"Korean\": \"Asian\", \"Chinese\": \"Asian\"}\n",
    ")\n",
    "specimens_df[\"ethnicity_condensed\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False    16\n",
       "Name: ethnicity_condensed, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df[\"ethnicity_condensed\"].isna().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], Name: ethnicity, dtype: int64)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Are there any non-NaN ethnicity values that we did not remap?\n",
    "specimens_df[specimens_df[\"ethnicity_condensed\"].isna()][\"ethnicity\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study_id</th>\n",
       "      <th>patient_id_within_study</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>disease_subtype</th>\n",
       "      <th>disease</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>ethnicity_condensed</th>\n",
       "      <th>study_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d11</td>\n",
       "      <td>A_d11</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d17</td>\n",
       "      <td>A_d17</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d45</td>\n",
       "      <td>A_d45</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d10</td>\n",
       "      <td>B_d10</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d19</td>\n",
       "      <td>B_d19</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d6</td>\n",
       "      <td>C_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d15</td>\n",
       "      <td>C_d15</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d6</td>\n",
       "      <td>D_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d28</td>\n",
       "      <td>D_d28</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d23</td>\n",
       "      <td>E_d23</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d44</td>\n",
       "      <td>E_d44</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d99</td>\n",
       "      <td>E_d99</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d14</td>\n",
       "      <td>F_d14</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d36</td>\n",
       "      <td>F_d36</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d9</td>\n",
       "      <td>G_d9</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d22</td>\n",
       "      <td>G_d22</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id     study_id patient_id_within_study sex   age ethnicity disease_subtype   disease timepoint specimen_label ethnicity_condensed study_name\n",
       "100  5f21e814e1adeb2edc12613c  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19       d11          A_d11               Asian        Kim\n",
       "101  5f21e814e1adeb2edc12613d  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19       d17          A_d17               Asian        Kim\n",
       "102  5f21e815e1adeb2edc12613e  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19       d45          A_d45               Asian        Kim\n",
       "103  5f21e815e1adeb2edc12613f  PRJNA648677                       B   M  55.0    Korean            Case  COVID-19       d10          B_d10               Asian        Kim\n",
       "104  5f21e815e1adeb2edc126140  PRJNA648677                       B   M  55.0    Korean            Case  COVID-19       d19          B_d19               Asian        Kim\n",
       "105  5f21e816e1adeb2edc126141  PRJNA648677                       C   F  53.0    Korean            Case  COVID-19        d6           C_d6               Asian        Kim\n",
       "106  5f21e816e1adeb2edc126142  PRJNA648677                       C   F  53.0    Korean            Case  COVID-19       d15          C_d15               Asian        Kim\n",
       "107  5f21e816e1adeb2edc126143  PRJNA648677                       D   M  24.0    Korean            Case  COVID-19        d6           D_d6               Asian        Kim\n",
       "108  5f21e817e1adeb2edc126144  PRJNA648677                       D   M  24.0    Korean            Case  COVID-19       d28          D_d28               Asian        Kim\n",
       "109  5f21e817e1adeb2edc126145  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19       d23          E_d23               Asian        Kim\n",
       "110  5f21e817e1adeb2edc126146  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19       d44          E_d44               Asian        Kim\n",
       "111  5f21e818e1adeb2edc126147  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19       d99          E_d99               Asian        Kim\n",
       "112  5f21e818e1adeb2edc126148  PRJNA648677                       F   F  40.0   Chinese            Case  COVID-19       d14          F_d14               Asian        Kim\n",
       "113  5f21e818e1adeb2edc126149  PRJNA648677                       F   F  40.0   Chinese            Case  COVID-19       d36          F_d36               Asian        Kim\n",
       "114  5f21e818e1adeb2edc12614a  PRJNA648677                       G   F  59.0    Korean            Case  COVID-19        d9           G_d9               Asian        Kim\n",
       "115  5f21e819e1adeb2edc12614b  PRJNA648677                       G   F  59.0    Korean            Case  COVID-19       d22          G_d22               Asian        Kim"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df[\"study_name\"] = specimens_df[\"study_id\"].replace(study_names)\n",
    "specimens_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study_id</th>\n",
       "      <th>patient_id_within_study</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>disease_subtype</th>\n",
       "      <th>disease</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>ethnicity_condensed</th>\n",
       "      <th>study_name</th>\n",
       "      <th>participant_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d11</td>\n",
       "      <td>A_d11</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d17</td>\n",
       "      <td>A_d17</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d45</td>\n",
       "      <td>A_d45</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d10</td>\n",
       "      <td>B_d10</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d19</td>\n",
       "      <td>B_d19</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d6</td>\n",
       "      <td>C_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d15</td>\n",
       "      <td>C_d15</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d6</td>\n",
       "      <td>D_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d28</td>\n",
       "      <td>D_d28</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d23</td>\n",
       "      <td>E_d23</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d44</td>\n",
       "      <td>E_d44</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d99</td>\n",
       "      <td>E_d99</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d14</td>\n",
       "      <td>F_d14</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d36</td>\n",
       "      <td>F_d36</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d9</td>\n",
       "      <td>G_d9</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>d22</td>\n",
       "      <td>G_d22</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id     study_id patient_id_within_study sex   age ethnicity disease_subtype   disease timepoint specimen_label ethnicity_condensed study_name participant_label\n",
       "100  5f21e814e1adeb2edc12613c  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19       d11          A_d11               Asian        Kim             Kim_A\n",
       "101  5f21e814e1adeb2edc12613d  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19       d17          A_d17               Asian        Kim             Kim_A\n",
       "102  5f21e815e1adeb2edc12613e  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19       d45          A_d45               Asian        Kim             Kim_A\n",
       "103  5f21e815e1adeb2edc12613f  PRJNA648677                       B   M  55.0    Korean            Case  COVID-19       d10          B_d10               Asian        Kim             Kim_B\n",
       "104  5f21e815e1adeb2edc126140  PRJNA648677                       B   M  55.0    Korean            Case  COVID-19       d19          B_d19               Asian        Kim             Kim_B\n",
       "105  5f21e816e1adeb2edc126141  PRJNA648677                       C   F  53.0    Korean            Case  COVID-19        d6           C_d6               Asian        Kim             Kim_C\n",
       "106  5f21e816e1adeb2edc126142  PRJNA648677                       C   F  53.0    Korean            Case  COVID-19       d15          C_d15               Asian        Kim             Kim_C\n",
       "107  5f21e816e1adeb2edc126143  PRJNA648677                       D   M  24.0    Korean            Case  COVID-19        d6           D_d6               Asian        Kim             Kim_D\n",
       "108  5f21e817e1adeb2edc126144  PRJNA648677                       D   M  24.0    Korean            Case  COVID-19       d28          D_d28               Asian        Kim             Kim_D\n",
       "109  5f21e817e1adeb2edc126145  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19       d23          E_d23               Asian        Kim             Kim_E\n",
       "110  5f21e817e1adeb2edc126146  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19       d44          E_d44               Asian        Kim             Kim_E\n",
       "111  5f21e818e1adeb2edc126147  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19       d99          E_d99               Asian        Kim             Kim_E\n",
       "112  5f21e818e1adeb2edc126148  PRJNA648677                       F   F  40.0   Chinese            Case  COVID-19       d14          F_d14               Asian        Kim             Kim_F\n",
       "113  5f21e818e1adeb2edc126149  PRJNA648677                       F   F  40.0   Chinese            Case  COVID-19       d36          F_d36               Asian        Kim             Kim_F\n",
       "114  5f21e818e1adeb2edc12614a  PRJNA648677                       G   F  59.0    Korean            Case  COVID-19        d9           G_d9               Asian        Kim             Kim_G\n",
       "115  5f21e819e1adeb2edc12614b  PRJNA648677                       G   F  59.0    Korean            Case  COVID-19       d22          G_d22               Asian        Kim             Kim_G"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df[\"participant_label\"] = (\n",
    "    specimens_df[\"study_name\"].str.strip()\n",
    "    + \"_\"\n",
    "    + specimens_df[\"patient_id_within_study\"].str.strip()\n",
    ")\n",
    "specimens_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study_id</th>\n",
       "      <th>patient_id_within_study</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>disease_subtype</th>\n",
       "      <th>disease</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>ethnicity_condensed</th>\n",
       "      <th>study_name</th>\n",
       "      <th>participant_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>11</td>\n",
       "      <td>A_d11</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>17</td>\n",
       "      <td>A_d17</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>45</td>\n",
       "      <td>A_d45</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>10</td>\n",
       "      <td>B_d10</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>19</td>\n",
       "      <td>B_d19</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>6</td>\n",
       "      <td>C_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>15</td>\n",
       "      <td>C_d15</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>6</td>\n",
       "      <td>D_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>28</td>\n",
       "      <td>D_d28</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>23</td>\n",
       "      <td>E_d23</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>44</td>\n",
       "      <td>E_d44</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>99</td>\n",
       "      <td>E_d99</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>14</td>\n",
       "      <td>F_d14</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>36</td>\n",
       "      <td>F_d36</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>9</td>\n",
       "      <td>G_d9</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>COVID-19</td>\n",
       "      <td>22</td>\n",
       "      <td>G_d22</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id     study_id patient_id_within_study sex   age ethnicity disease_subtype   disease  timepoint specimen_label ethnicity_condensed study_name participant_label\n",
       "100  5f21e814e1adeb2edc12613c  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19         11          A_d11               Asian        Kim             Kim_A\n",
       "101  5f21e814e1adeb2edc12613d  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19         17          A_d17               Asian        Kim             Kim_A\n",
       "102  5f21e815e1adeb2edc12613e  PRJNA648677                       A   M  55.0    Korean            Case  COVID-19         45          A_d45               Asian        Kim             Kim_A\n",
       "103  5f21e815e1adeb2edc12613f  PRJNA648677                       B   M  55.0    Korean            Case  COVID-19         10          B_d10               Asian        Kim             Kim_B\n",
       "104  5f21e815e1adeb2edc126140  PRJNA648677                       B   M  55.0    Korean            Case  COVID-19         19          B_d19               Asian        Kim             Kim_B\n",
       "105  5f21e816e1adeb2edc126141  PRJNA648677                       C   F  53.0    Korean            Case  COVID-19          6           C_d6               Asian        Kim             Kim_C\n",
       "106  5f21e816e1adeb2edc126142  PRJNA648677                       C   F  53.0    Korean            Case  COVID-19         15          C_d15               Asian        Kim             Kim_C\n",
       "107  5f21e816e1adeb2edc126143  PRJNA648677                       D   M  24.0    Korean            Case  COVID-19          6           D_d6               Asian        Kim             Kim_D\n",
       "108  5f21e817e1adeb2edc126144  PRJNA648677                       D   M  24.0    Korean            Case  COVID-19         28          D_d28               Asian        Kim             Kim_D\n",
       "109  5f21e817e1adeb2edc126145  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19         23          E_d23               Asian        Kim             Kim_E\n",
       "110  5f21e817e1adeb2edc126146  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19         44          E_d44               Asian        Kim             Kim_E\n",
       "111  5f21e818e1adeb2edc126147  PRJNA648677                       E   M  48.0   Chinese            Case  COVID-19         99          E_d99               Asian        Kim             Kim_E\n",
       "112  5f21e818e1adeb2edc126148  PRJNA648677                       F   F  40.0   Chinese            Case  COVID-19         14          F_d14               Asian        Kim             Kim_F\n",
       "113  5f21e818e1adeb2edc126149  PRJNA648677                       F   F  40.0   Chinese            Case  COVID-19         36          F_d36               Asian        Kim             Kim_F\n",
       "114  5f21e818e1adeb2edc12614a  PRJNA648677                       G   F  59.0    Korean            Case  COVID-19          9           G_d9               Asian        Kim             Kim_G\n",
       "115  5f21e819e1adeb2edc12614b  PRJNA648677                       G   F  59.0    Korean            Case  COVID-19         22          G_d22               Asian        Kim             Kim_G"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# extract number\n",
    "specimens_df[\"timepoint\"] = specimens_df[\"timepoint\"].str.extract(\"(\\d+)\").astype(int)\n",
    "specimens_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "specimens_df[\"disease\"] = specimens_df[\"disease\"].replace({\"COVID-19\": \"Covid19\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((16, 13), 7)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df.shape, specimens_df[\"participant_label\"].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Look at timepoints, decide which ones are peak"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study_id</th>\n",
       "      <th>patient_id_within_study</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>disease_subtype</th>\n",
       "      <th>disease</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>ethnicity_condensed</th>\n",
       "      <th>study_name</th>\n",
       "      <th>participant_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>11</td>\n",
       "      <td>A_d11</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>17</td>\n",
       "      <td>A_d17</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>45</td>\n",
       "      <td>A_d45</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>10</td>\n",
       "      <td>B_d10</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>19</td>\n",
       "      <td>B_d19</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>6</td>\n",
       "      <td>C_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>15</td>\n",
       "      <td>C_d15</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>6</td>\n",
       "      <td>D_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>28</td>\n",
       "      <td>D_d28</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>23</td>\n",
       "      <td>E_d23</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>44</td>\n",
       "      <td>E_d44</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>99</td>\n",
       "      <td>E_d99</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>14</td>\n",
       "      <td>F_d14</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>36</td>\n",
       "      <td>F_d36</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>9</td>\n",
       "      <td>G_d9</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>22</td>\n",
       "      <td>G_d22</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id     study_id patient_id_within_study sex   age ethnicity disease_subtype  disease  timepoint specimen_label ethnicity_condensed study_name participant_label\n",
       "100  5f21e814e1adeb2edc12613c  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         11          A_d11               Asian        Kim             Kim_A\n",
       "101  5f21e814e1adeb2edc12613d  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         17          A_d17               Asian        Kim             Kim_A\n",
       "102  5f21e815e1adeb2edc12613e  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         45          A_d45               Asian        Kim             Kim_A\n",
       "103  5f21e815e1adeb2edc12613f  PRJNA648677                       B   M  55.0    Korean            Case  Covid19         10          B_d10               Asian        Kim             Kim_B\n",
       "104  5f21e815e1adeb2edc126140  PRJNA648677                       B   M  55.0    Korean            Case  Covid19         19          B_d19               Asian        Kim             Kim_B\n",
       "105  5f21e816e1adeb2edc126141  PRJNA648677                       C   F  53.0    Korean            Case  Covid19          6           C_d6               Asian        Kim             Kim_C\n",
       "106  5f21e816e1adeb2edc126142  PRJNA648677                       C   F  53.0    Korean            Case  Covid19         15          C_d15               Asian        Kim             Kim_C\n",
       "107  5f21e816e1adeb2edc126143  PRJNA648677                       D   M  24.0    Korean            Case  Covid19          6           D_d6               Asian        Kim             Kim_D\n",
       "108  5f21e817e1adeb2edc126144  PRJNA648677                       D   M  24.0    Korean            Case  Covid19         28          D_d28               Asian        Kim             Kim_D\n",
       "109  5f21e817e1adeb2edc126145  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         23          E_d23               Asian        Kim             Kim_E\n",
       "110  5f21e817e1adeb2edc126146  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         44          E_d44               Asian        Kim             Kim_E\n",
       "111  5f21e818e1adeb2edc126147  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         99          E_d99               Asian        Kim             Kim_E\n",
       "112  5f21e818e1adeb2edc126148  PRJNA648677                       F   F  40.0   Chinese            Case  Covid19         14          F_d14               Asian        Kim             Kim_F\n",
       "113  5f21e818e1adeb2edc126149  PRJNA648677                       F   F  40.0   Chinese            Case  Covid19         36          F_d36               Asian        Kim             Kim_F\n",
       "114  5f21e818e1adeb2edc12614a  PRJNA648677                       G   F  59.0    Korean            Case  Covid19          9           G_d9               Asian        Kim             Kim_G\n",
       "115  5f21e819e1adeb2edc12614b  PRJNA648677                       G   F  59.0    Korean            Case  Covid19         22          G_d22               Asian        Kim             Kim_G"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "participant_label\n",
       "Kim_A    3\n",
       "Kim_E    3\n",
       "Kim_B    2\n",
       "Kim_C    2\n",
       "Kim_D    2\n",
       "Kim_F    2\n",
       "Kim_G    2\n",
       "dtype: int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df.groupby(\"participant_label\").size().sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>num_replicates</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>participant_label</th>\n",
       "      <th>timepoint</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">Kim_A</th>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Kim_B</th>\n",
       "      <th>10</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             num_replicates\n",
       "participant_label timepoint                \n",
       "Kim_A             11                      1\n",
       "                  17                      1\n",
       "                  45                      1\n",
       "Kim_B             10                      1\n",
       "                  19                      1"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df.groupby([\"participant_label\", \"timepoint\"]).size().sort_values(\n",
    "    ascending=False\n",
    ").to_frame(\"num_replicates\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # can't do it this way because this will choose only one row as peak per patient, whereas we want all replicates from the peak timepoint to be marked as peak\n",
    "# specimens_df['is_peak'] = False\n",
    "# specimens_df.loc[specimens_df.groupby(\"participant_label\", observed=True)[\"timepoint\"].idxmax(), 'is_peak'] = True\n",
    "# specimens_df['is_peak'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study_id</th>\n",
       "      <th>patient_id_within_study</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>disease_subtype</th>\n",
       "      <th>disease</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>ethnicity_condensed</th>\n",
       "      <th>study_name</th>\n",
       "      <th>participant_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>11</td>\n",
       "      <td>A_d11</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>17</td>\n",
       "      <td>A_d17</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>45</td>\n",
       "      <td>A_d45</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>10</td>\n",
       "      <td>B_d10</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>19</td>\n",
       "      <td>B_d19</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>6</td>\n",
       "      <td>C_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>15</td>\n",
       "      <td>C_d15</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>6</td>\n",
       "      <td>D_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>28</td>\n",
       "      <td>D_d28</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>23</td>\n",
       "      <td>E_d23</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>44</td>\n",
       "      <td>E_d44</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>99</td>\n",
       "      <td>E_d99</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>14</td>\n",
       "      <td>F_d14</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>36</td>\n",
       "      <td>F_d36</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>9</td>\n",
       "      <td>G_d9</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>22</td>\n",
       "      <td>G_d22</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                repertoire_id     study_id patient_id_within_study sex   age ethnicity disease_subtype  disease  timepoint specimen_label ethnicity_condensed study_name participant_label\n",
       "100  5f21e814e1adeb2edc12613c  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         11          A_d11               Asian        Kim             Kim_A\n",
       "101  5f21e814e1adeb2edc12613d  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         17          A_d17               Asian        Kim             Kim_A\n",
       "102  5f21e815e1adeb2edc12613e  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         45          A_d45               Asian        Kim             Kim_A\n",
       "103  5f21e815e1adeb2edc12613f  PRJNA648677                       B   M  55.0    Korean            Case  Covid19         10          B_d10               Asian        Kim             Kim_B\n",
       "104  5f21e815e1adeb2edc126140  PRJNA648677                       B   M  55.0    Korean            Case  Covid19         19          B_d19               Asian        Kim             Kim_B\n",
       "105  5f21e816e1adeb2edc126141  PRJNA648677                       C   F  53.0    Korean            Case  Covid19          6           C_d6               Asian        Kim             Kim_C\n",
       "106  5f21e816e1adeb2edc126142  PRJNA648677                       C   F  53.0    Korean            Case  Covid19         15          C_d15               Asian        Kim             Kim_C\n",
       "107  5f21e816e1adeb2edc126143  PRJNA648677                       D   M  24.0    Korean            Case  Covid19          6           D_d6               Asian        Kim             Kim_D\n",
       "108  5f21e817e1adeb2edc126144  PRJNA648677                       D   M  24.0    Korean            Case  Covid19         28          D_d28               Asian        Kim             Kim_D\n",
       "109  5f21e817e1adeb2edc126145  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         23          E_d23               Asian        Kim             Kim_E\n",
       "110  5f21e817e1adeb2edc126146  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         44          E_d44               Asian        Kim             Kim_E\n",
       "111  5f21e818e1adeb2edc126147  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         99          E_d99               Asian        Kim             Kim_E\n",
       "112  5f21e818e1adeb2edc126148  PRJNA648677                       F   F  40.0   Chinese            Case  Covid19         14          F_d14               Asian        Kim             Kim_F\n",
       "113  5f21e818e1adeb2edc126149  PRJNA648677                       F   F  40.0   Chinese            Case  Covid19         36          F_d36               Asian        Kim             Kim_F\n",
       "114  5f21e818e1adeb2edc12614a  PRJNA648677                       G   F  59.0    Korean            Case  Covid19          9           G_d9               Asian        Kim             Kim_G\n",
       "115  5f21e819e1adeb2edc12614b  PRJNA648677                       G   F  59.0    Korean            Case  Covid19         22          G_d22               Asian        Kim             Kim_G"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimens_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study_id</th>\n",
       "      <th>patient_id_within_study</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>disease_subtype</th>\n",
       "      <th>disease</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>ethnicity_condensed</th>\n",
       "      <th>study_name</th>\n",
       "      <th>participant_label</th>\n",
       "      <th>timepoint_diff_from_15</th>\n",
       "      <th>is_peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>17</td>\n",
       "      <td>A_d17</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>19</td>\n",
       "      <td>B_d19</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>15</td>\n",
       "      <td>C_d15</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>28</td>\n",
       "      <td>D_d28</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "      <td>13</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>23</td>\n",
       "      <td>E_d23</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>14</td>\n",
       "      <td>F_d14</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>22</td>\n",
       "      <td>G_d22</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "      <td>7</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               repertoire_id     study_id patient_id_within_study sex   age ethnicity disease_subtype  disease  timepoint specimen_label ethnicity_condensed study_name participant_label  timepoint_diff_from_15  is_peak\n",
       "1   5f21e814e1adeb2edc12613d  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         17          A_d17               Asian        Kim             Kim_A                       2     True\n",
       "4   5f21e815e1adeb2edc126140  PRJNA648677                       B   M  55.0    Korean            Case  Covid19         19          B_d19               Asian        Kim             Kim_B                       4     True\n",
       "5   5f21e816e1adeb2edc126142  PRJNA648677                       C   F  53.0    Korean            Case  Covid19         15          C_d15               Asian        Kim             Kim_C                       0     True\n",
       "6   5f21e817e1adeb2edc126144  PRJNA648677                       D   M  24.0    Korean            Case  Covid19         28          D_d28               Asian        Kim             Kim_D                      13     True\n",
       "7   5f21e817e1adeb2edc126145  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         23          E_d23               Asian        Kim             Kim_E                       8     True\n",
       "9   5f21e818e1adeb2edc126148  PRJNA648677                       F   F  40.0   Chinese            Case  Covid19         14          F_d14               Asian        Kim             Kim_F                       1     True\n",
       "11  5f21e819e1adeb2edc12614b  PRJNA648677                       G   F  59.0    Korean            Case  Covid19         22          G_d22               Asian        Kim             Kim_G                       7     True"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# choose peak timepoints per patient, with constraints on the timepoint range\n",
    "# and exclude patients known to have mild disease (e.g. from Montague et al study, subjects 1-2 are mild disease)\n",
    "\n",
    "# reset index to make sure .loc[idxmin] works properly\n",
    "peak_timepoint_per_patient = (\n",
    "    specimens_df[\n",
    "        (specimens_df[\"timepoint\"] >= 10)\n",
    "        & (specimens_df[\"timepoint\"] <= 45)\n",
    "        & (specimens_df[\"disease_subtype\"] != \"Mild\")\n",
    "    ]\n",
    ").reset_index(drop=True)\n",
    "\n",
    "\n",
    "# choose the timepoint closest to day 15\n",
    "# choose one row per group\n",
    "peak_timepoint_per_patient[\"timepoint_diff_from_15\"] = (\n",
    "    peak_timepoint_per_patient[\"timepoint\"] - 15\n",
    ").abs()\n",
    "peak_timepoint_per_patient = peak_timepoint_per_patient.loc[\n",
    "    peak_timepoint_per_patient.groupby(\"participant_label\", observed=True)[\n",
    "        \"timepoint_diff_from_15\"\n",
    "    ].idxmin()\n",
    "].assign(is_peak=True)\n",
    "peak_timepoint_per_patient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>repertoire_id</th>\n",
       "      <th>study_id</th>\n",
       "      <th>patient_id_within_study</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>disease_subtype</th>\n",
       "      <th>disease</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>ethnicity_condensed</th>\n",
       "      <th>study_name</th>\n",
       "      <th>participant_label</th>\n",
       "      <th>is_peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5f21e814e1adeb2edc12613c</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>11</td>\n",
       "      <td>A_d11</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5f21e814e1adeb2edc12613d</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>17</td>\n",
       "      <td>A_d17</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5f21e815e1adeb2edc12613e</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>45</td>\n",
       "      <td>A_d45</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_A</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5f21e815e1adeb2edc12613f</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>10</td>\n",
       "      <td>B_d10</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5f21e815e1adeb2edc126140</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>19</td>\n",
       "      <td>B_d19</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_B</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5f21e816e1adeb2edc126141</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>6</td>\n",
       "      <td>C_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5f21e816e1adeb2edc126142</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>15</td>\n",
       "      <td>C_d15</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_C</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5f21e816e1adeb2edc126143</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>6</td>\n",
       "      <td>D_d6</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5f21e817e1adeb2edc126144</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>28</td>\n",
       "      <td>D_d28</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_D</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5f21e817e1adeb2edc126145</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>23</td>\n",
       "      <td>E_d23</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5f21e817e1adeb2edc126146</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>44</td>\n",
       "      <td>E_d44</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5f21e818e1adeb2edc126147</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>99</td>\n",
       "      <td>E_d99</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_E</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5f21e818e1adeb2edc126148</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>14</td>\n",
       "      <td>F_d14</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5f21e818e1adeb2edc126149</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Chinese</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>36</td>\n",
       "      <td>F_d36</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_F</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5f21e818e1adeb2edc12614a</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>9</td>\n",
       "      <td>G_d9</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5f21e819e1adeb2edc12614b</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Korean</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>22</td>\n",
       "      <td>G_d22</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Kim</td>\n",
       "      <td>Kim_G</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               repertoire_id     study_id patient_id_within_study sex   age ethnicity disease_subtype  disease  timepoint specimen_label ethnicity_condensed study_name participant_label  is_peak\n",
       "0   5f21e814e1adeb2edc12613c  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         11          A_d11               Asian        Kim             Kim_A    False\n",
       "1   5f21e814e1adeb2edc12613d  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         17          A_d17               Asian        Kim             Kim_A     True\n",
       "2   5f21e815e1adeb2edc12613e  PRJNA648677                       A   M  55.0    Korean            Case  Covid19         45          A_d45               Asian        Kim             Kim_A    False\n",
       "3   5f21e815e1adeb2edc12613f  PRJNA648677                       B   M  55.0    Korean            Case  Covid19         10          B_d10               Asian        Kim             Kim_B    False\n",
       "4   5f21e815e1adeb2edc126140  PRJNA648677                       B   M  55.0    Korean            Case  Covid19         19          B_d19               Asian        Kim             Kim_B     True\n",
       "5   5f21e816e1adeb2edc126141  PRJNA648677                       C   F  53.0    Korean            Case  Covid19          6           C_d6               Asian        Kim             Kim_C    False\n",
       "6   5f21e816e1adeb2edc126142  PRJNA648677                       C   F  53.0    Korean            Case  Covid19         15          C_d15               Asian        Kim             Kim_C     True\n",
       "7   5f21e816e1adeb2edc126143  PRJNA648677                       D   M  24.0    Korean            Case  Covid19          6           D_d6               Asian        Kim             Kim_D    False\n",
       "8   5f21e817e1adeb2edc126144  PRJNA648677                       D   M  24.0    Korean            Case  Covid19         28          D_d28               Asian        Kim             Kim_D     True\n",
       "9   5f21e817e1adeb2edc126145  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         23          E_d23               Asian        Kim             Kim_E     True\n",
       "10  5f21e817e1adeb2edc126146  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         44          E_d44               Asian        Kim             Kim_E    False\n",
       "11  5f21e818e1adeb2edc126147  PRJNA648677                       E   M  48.0   Chinese            Case  Covid19         99          E_d99               Asian        Kim             Kim_E    False\n",
       "12  5f21e818e1adeb2edc126148  PRJNA648677                       F   F  40.0   Chinese            Case  Covid19         14          F_d14               Asian        Kim             Kim_F     True\n",
       "13  5f21e818e1adeb2edc126149  PRJNA648677                       F   F  40.0   Chinese            Case  Covid19         36          F_d36               Asian        Kim             Kim_F    False\n",
       "14  5f21e818e1adeb2edc12614a  PRJNA648677                       G   F  59.0    Korean            Case  Covid19          9           G_d9               Asian        Kim             Kim_G    False\n",
       "15  5f21e819e1adeb2edc12614b  PRJNA648677                       G   F  59.0    Korean            Case  Covid19         22          G_d22               Asian        Kim             Kim_G     True"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Note that peak timepoint may have many replicates!\n",
    "specimens_df2 = pd.merge(\n",
    "    specimens_df,\n",
    "    peak_timepoint_per_patient[[\"participant_label\", \"timepoint\", \"is_peak\"]],\n",
    "    on=[\"participant_label\", \"timepoint\"],\n",
    "    how=\"left\",\n",
    ")\n",
    "specimens_df2[\"is_peak\"].fillna(False, inplace=True)\n",
    "assert specimens_df2.shape[0] == specimens_df.shape[0]\n",
    "specimens_df = specimens_df2\n",
    "specimens_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7, 7)"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# not all patients have any peak timepoints chosen\n",
    "specimens_df[\"participant_label\"].nunique(), specimens_df[specimens_df[\"is_peak\"]][\n",
    "    \"participant_label\"\n",
    "].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# not all patients have any peak timepoints chosen\n",
    "set(specimens_df[\"participant_label\"]) - set(\n",
    "    specimens_df[specimens_df[\"is_peak\"]][\"participant_label\"]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "participant_label  timepoint\n",
       "Kim_A              17           1\n",
       "Kim_B              19           1\n",
       "Kim_C              15           1\n",
       "Kim_D              28           1\n",
       "Kim_E              23           1\n",
       "Kim_F              14           1\n",
       "Kim_G              22           1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# how many replicates chosen as peak per patient (should be more than 1 replicate for many)\n",
    "specimens_df[specimens_df[\"is_peak\"]].groupby(\n",
    "    [\"participant_label\", \"timepoint\"]\n",
    ").size().sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_label</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>is_peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Kim_A</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Kim_B</td>\n",
       "      <td>19</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Kim_C</td>\n",
       "      <td>15</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Kim_D</td>\n",
       "      <td>28</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Kim_E</td>\n",
       "      <td>23</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Kim_F</td>\n",
       "      <td>14</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Kim_G</td>\n",
       "      <td>22</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   participant_label  timepoint  is_peak\n",
       "1              Kim_A         17     True\n",
       "4              Kim_B         19     True\n",
       "6              Kim_C         15     True\n",
       "8              Kim_D         28     True\n",
       "9              Kim_E         23     True\n",
       "12             Kim_F         14     True\n",
       "15             Kim_G         22     True"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# which were chosen\n",
    "specimens_df[specimens_df[\"is_peak\"]][\n",
    "    [\"participant_label\", \"timepoint\", \"is_peak\"]\n",
    "].sort_values([\"participant_label\", \"timepoint\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>participant_label</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>is_peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A_d11</td>\n",
       "      <td>Kim_A</td>\n",
       "      <td>11</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A_d17</td>\n",
       "      <td>Kim_A</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A_d45</td>\n",
       "      <td>Kim_A</td>\n",
       "      <td>45</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>B_d10</td>\n",
       "      <td>Kim_B</td>\n",
       "      <td>10</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>B_d19</td>\n",
       "      <td>Kim_B</td>\n",
       "      <td>19</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C_d6</td>\n",
       "      <td>Kim_C</td>\n",
       "      <td>6</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>C_d15</td>\n",
       "      <td>Kim_C</td>\n",
       "      <td>15</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>D_d6</td>\n",
       "      <td>Kim_D</td>\n",
       "      <td>6</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>D_d28</td>\n",
       "      <td>Kim_D</td>\n",
       "      <td>28</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>E_d23</td>\n",
       "      <td>Kim_E</td>\n",
       "      <td>23</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>E_d44</td>\n",
       "      <td>Kim_E</td>\n",
       "      <td>44</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>E_d99</td>\n",
       "      <td>Kim_E</td>\n",
       "      <td>99</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>F_d14</td>\n",
       "      <td>Kim_F</td>\n",
       "      <td>14</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>F_d36</td>\n",
       "      <td>Kim_F</td>\n",
       "      <td>36</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>G_d9</td>\n",
       "      <td>Kim_G</td>\n",
       "      <td>9</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>G_d22</td>\n",
       "      <td>Kim_G</td>\n",
       "      <td>22</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   specimen_label participant_label  timepoint  is_peak\n",
       "0           A_d11             Kim_A         11    False\n",
       "1           A_d17             Kim_A         17     True\n",
       "2           A_d45             Kim_A         45    False\n",
       "3           B_d10             Kim_B         10    False\n",
       "4           B_d19             Kim_B         19     True\n",
       "5            C_d6             Kim_C          6    False\n",
       "6           C_d15             Kim_C         15     True\n",
       "7            D_d6             Kim_D          6    False\n",
       "8           D_d28             Kim_D         28     True\n",
       "9           E_d23             Kim_E         23     True\n",
       "10          E_d44             Kim_E         44    False\n",
       "11          E_d99             Kim_E         99    False\n",
       "12          F_d14             Kim_F         14     True\n",
       "13          F_d36             Kim_F         36    False\n",
       "14           G_d9             Kim_G          9    False\n",
       "15          G_d22             Kim_G         22     True"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimen_metadata_extra = (\n",
    "    specimens_df[[\"specimen_label\", \"participant_label\", \"timepoint\", \"is_peak\"]]\n",
    "    .drop_duplicates()\n",
    "    .sort_values([\"participant_label\", \"timepoint\"])\n",
    ")\n",
    "specimen_metadata_extra.to_csv(\n",
    "    config.paths.metadata_dir\n",
    "    / \"generated.external_cohorts.covid19_bcr.specimen_metadata_extra.tsv\",\n",
    "    sep=\"\\t\",\n",
    "    index=None,\n",
    ")\n",
    "specimen_metadata_extra"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_label</th>\n",
       "      <th>study_id</th>\n",
       "      <th>patient_id_within_study</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity_condensed</th>\n",
       "      <th>disease_subtype</th>\n",
       "      <th>disease</th>\n",
       "      <th>study_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Kim_A</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>A</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Kim_B</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>B</td>\n",
       "      <td>M</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Kim_C</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>C</td>\n",
       "      <td>F</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Kim_D</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>D</td>\n",
       "      <td>M</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Kim_E</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>E</td>\n",
       "      <td>M</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Kim_F</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Kim_G</td>\n",
       "      <td>PRJNA648677</td>\n",
       "      <td>G</td>\n",
       "      <td>F</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Asian</td>\n",
       "      <td>Case</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   participant_label     study_id patient_id_within_study sex   age ethnicity_condensed disease_subtype  disease study_name\n",
       "0              Kim_A  PRJNA648677                       A   M  55.0               Asian            Case  Covid19        Kim\n",
       "3              Kim_B  PRJNA648677                       B   M  55.0               Asian            Case  Covid19        Kim\n",
       "5              Kim_C  PRJNA648677                       C   F  53.0               Asian            Case  Covid19        Kim\n",
       "7              Kim_D  PRJNA648677                       D   M  24.0               Asian            Case  Covid19        Kim\n",
       "9              Kim_E  PRJNA648677                       E   M  48.0               Asian            Case  Covid19        Kim\n",
       "12             Kim_F  PRJNA648677                       F   F  40.0               Asian            Case  Covid19        Kim\n",
       "14             Kim_G  PRJNA648677                       G   F  59.0               Asian            Case  Covid19        Kim"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "participant_df = specimens_df[\n",
    "    [\n",
    "        \"participant_label\",\n",
    "        \"study_id\",\n",
    "        \"patient_id_within_study\",\n",
    "        \"sex\",\n",
    "        \"age\",\n",
    "        \"ethnicity_condensed\",\n",
    "        \"disease_subtype\",\n",
    "        \"disease\",\n",
    "        \"study_name\",\n",
    "    ]\n",
    "].drop_duplicates()\n",
    "participant_df.to_csv(\n",
    "    config.paths.metadata_dir\n",
    "    / \"generated.external_cohorts.covid19_bcr.participant_metadata.tsv\",\n",
    "    sep=\"\\t\",\n",
    "    index=None,\n",
    ")\n",
    "participant_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((7, 9), (16, 4))"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "participant_df.shape, specimen_metadata_extra.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# now for briney-healthy\n",
    "\n",
    "we already have `patient_id` in `ireceptor_data.briney_healthy_sequences`\n",
    "\n",
    "soon we will add more replicates i.e. more samples though. see `repertoire_id` TODO comments.\n",
    "\n",
    "for now, just add some simple metadata to create an \"all external participants\" metadata file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(8, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>participant_label</th>\n",
       "      <th>study_name</th>\n",
       "      <th>disease</th>\n",
       "      <th>is_peak</th>\n",
       "      <th>timepoint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>D103_1</td>\n",
       "      <td>D103</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>326780_1</td>\n",
       "      <td>326780</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>326650_1</td>\n",
       "      <td>326650</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>326737_1</td>\n",
       "      <td>326737</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>327059_1</td>\n",
       "      <td>327059</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>326907_1</td>\n",
       "      <td>326907</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>316188_1</td>\n",
       "      <td>316188</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>326797_1</td>\n",
       "      <td>326797</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  specimen_label participant_label study_name             disease  is_peak  timepoint\n",
       "0         D103_1              D103     Briney  Healthy/Background     True          0\n",
       "1       326780_1            326780     Briney  Healthy/Background     True          0\n",
       "2       326650_1            326650     Briney  Healthy/Background     True          0\n",
       "3       326737_1            326737     Briney  Healthy/Background     True          0\n",
       "4       327059_1            327059     Briney  Healthy/Background     True          0\n",
       "5       326907_1            326907     Briney  Healthy/Background     True          0\n",
       "6       316188_1            316188     Briney  Healthy/Background     True          0\n",
       "7       326797_1            326797     Briney  Healthy/Background     True          0"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "briney_patients = pd.DataFrame(\n",
    "    {\n",
    "        \"specimen_label\": [\n",
    "            \"D103_1\",\n",
    "            \"326780_1\",\n",
    "            \"326650_1\",\n",
    "            \"326737_1\",\n",
    "            \"327059_1\",\n",
    "            \"326907_1\",\n",
    "            \"316188_1\",\n",
    "            \"326797_1\",\n",
    "        ]\n",
    "    }\n",
    ")\n",
    "briney_patients[\"participant_label\"] = (\n",
    "    briney_patients[\"specimen_label\"].str.split(\"_\").str[0]\n",
    ")\n",
    "briney_patients[\"study_name\"] = \"Briney\"\n",
    "briney_patients[\"disease\"] = healthy_label\n",
    "# all healthy are \"peak\" and 0 timepoint\n",
    "briney_patients[\"is_peak\"] = True\n",
    "briney_patients[\"timepoint\"] = 0\n",
    "print(briney_patients.shape)\n",
    "briney_patients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(10, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>316188</td>\n",
       "      <td>30</td>\n",
       "      <td>F</td>\n",
       "      <td>African American</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>326650</td>\n",
       "      <td>18</td>\n",
       "      <td>F</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>326651</td>\n",
       "      <td>19</td>\n",
       "      <td>M</td>\n",
       "      <td>African American</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>326713</td>\n",
       "      <td>25</td>\n",
       "      <td>F</td>\n",
       "      <td>African American</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>326780</td>\n",
       "      <td>29</td>\n",
       "      <td>M</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>326797</td>\n",
       "      <td>21</td>\n",
       "      <td>F</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>326737</td>\n",
       "      <td>29</td>\n",
       "      <td>M</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>326907</td>\n",
       "      <td>29</td>\n",
       "      <td>F</td>\n",
       "      <td>African American</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>327059</td>\n",
       "      <td>26</td>\n",
       "      <td>M</td>\n",
       "      <td>African American / Caucasian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>D103</td>\n",
       "      <td>25</td>\n",
       "      <td>M</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subject  age sex                     ethnicity\n",
       "0  316188   30   F              African American\n",
       "1  326650   18   F                     Caucasian\n",
       "2  326651   19   M              African American\n",
       "3  326713   25   F              African American\n",
       "4  326780   29   M                     Caucasian\n",
       "5  326797   21   F                     Caucasian\n",
       "6  326737   29   M                     Caucasian\n",
       "7  326907   29   F              African American\n",
       "8  327059   26   M  African American / Caucasian\n",
       "9    D103   25   M                     Caucasian"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Original paper table is wrong: 326907 is listed twice with different values; 326737 is missing. One of the dupes should be 326737.\n",
    "# Fixed based on:\n",
    "# https://www.ncbi.nlm.nih.gov/biosample/10331432\n",
    "# https://www.ncbi.nlm.nih.gov/biosample/10331429\n",
    "briney_demographics = pd.read_csv(config.paths.metadata_dir / \"briney_demographics.csv\")\n",
    "assert not briney_demographics[\"subject\"].duplicated().any()\n",
    "print(briney_demographics.shape)\n",
    "briney_demographics[\"sex\"] = briney_demographics[\"sex\"].replace(\n",
    "    {\"male\": \"M\", \"female\": \"F\"}\n",
    ")\n",
    "briney_demographics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Caucasian                       5\n",
       "African American                4\n",
       "African American / Caucasian    1\n",
       "Name: ethnicity, dtype: int64"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "briney_demographics[\"ethnicity\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Caucasian    5\n",
       "African      4\n",
       "Name: ethnicity_condensed, dtype: int64"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "briney_demographics[\"ethnicity_condensed\"] = briney_demographics[\"ethnicity\"].replace(\n",
    "    {\"African American\": \"African\", \"African American / Caucasian\": np.nan}\n",
    ")\n",
    "briney_demographics[\"ethnicity_condensed\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False    9\n",
       "True     1\n",
       "Name: ethnicity_condensed, dtype: int64"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "briney_demographics[\"ethnicity_condensed\"].isna().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "African American / Caucasian    1\n",
       "Name: ethnicity, dtype: int64"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Are there any non-NaN ethnicity values that we did not remap?\n",
    "briney_demographics[briney_demographics[\"ethnicity_condensed\"].isna()][\n",
    "    \"ethnicity\"\n",
    "].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>specimen_label</th>\n",
       "      <th>participant_label</th>\n",
       "      <th>study_name</th>\n",
       "      <th>disease</th>\n",
       "      <th>is_peak</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>ethnicity_condensed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>D103_1</td>\n",
       "      <td>D103</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>M</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>326780_1</td>\n",
       "      <td>326780</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>M</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>326650_1</td>\n",
       "      <td>326650</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>F</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>326737_1</td>\n",
       "      <td>326737</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>M</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>327059_1</td>\n",
       "      <td>327059</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>M</td>\n",
       "      <td>African American / Caucasian</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>326907_1</td>\n",
       "      <td>326907</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>F</td>\n",
       "      <td>African American</td>\n",
       "      <td>African</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>316188_1</td>\n",
       "      <td>316188</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>F</td>\n",
       "      <td>African American</td>\n",
       "      <td>African</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>326797_1</td>\n",
       "      <td>326797</td>\n",
       "      <td>Briney</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>F</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>Caucasian</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  specimen_label participant_label study_name             disease  is_peak  timepoint  age sex                     ethnicity ethnicity_condensed\n",
       "0         D103_1              D103     Briney  Healthy/Background     True          0   25   M                     Caucasian           Caucasian\n",
       "1       326780_1            326780     Briney  Healthy/Background     True          0   29   M                     Caucasian           Caucasian\n",
       "2       326650_1            326650     Briney  Healthy/Background     True          0   18   F                     Caucasian           Caucasian\n",
       "3       326737_1            326737     Briney  Healthy/Background     True          0   29   M                     Caucasian           Caucasian\n",
       "4       327059_1            327059     Briney  Healthy/Background     True          0   26   M  African American / Caucasian                 NaN\n",
       "5       326907_1            326907     Briney  Healthy/Background     True          0   29   F              African American             African\n",
       "6       316188_1            316188     Briney  Healthy/Background     True          0   30   F              African American             African\n",
       "7       326797_1            326797     Briney  Healthy/Background     True          0   21   F                     Caucasian           Caucasian"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "briney_patients = pd.merge(\n",
    "    briney_patients,\n",
    "    briney_demographics.set_index(\"subject\"),\n",
    "    how=\"left\",\n",
    "    validate=\"1:1\",\n",
    "    left_on=\"participant_label\",\n",
    "    right_index=True,\n",
    ")\n",
    "briney_patients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "briney_patients.to_csv(\n",
    "    config.paths.metadata_dir\n",
    "    / \"generated.external_cohorts.healthy_bcr.participant_metadata.tsv\",\n",
    "    sep=\"\\t\",\n",
    "    index=None,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_label</th>\n",
       "      <th>disease</th>\n",
       "      <th>study_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Kim_A</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Kim_B</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Kim_C</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Kim_D</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Kim_E</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Kim_F</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Kim_G</td>\n",
       "      <td>Covid19</td>\n",
       "      <td>Kim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>D103</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>Briney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>326780</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>Briney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>326650</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>Briney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>326737</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>Briney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>327059</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>Briney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>326907</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>Briney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>316188</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>Briney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>326797</td>\n",
       "      <td>Healthy/Background</td>\n",
       "      <td>Briney</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   participant_label             disease study_name\n",
       "0              Kim_A             Covid19        Kim\n",
       "3              Kim_B             Covid19        Kim\n",
       "5              Kim_C             Covid19        Kim\n",
       "7              Kim_D             Covid19        Kim\n",
       "9              Kim_E             Covid19        Kim\n",
       "12             Kim_F             Covid19        Kim\n",
       "14             Kim_G             Covid19        Kim\n",
       "0               D103  Healthy/Background     Briney\n",
       "1             326780  Healthy/Background     Briney\n",
       "2             326650  Healthy/Background     Briney\n",
       "3             326737  Healthy/Background     Briney\n",
       "4             327059  Healthy/Background     Briney\n",
       "5             326907  Healthy/Background     Briney\n",
       "6             316188  Healthy/Background     Briney\n",
       "7             326797  Healthy/Background     Briney"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "participant_df_plus_briney = pd.concat(\n",
    "    [\n",
    "        participant_df[[\"participant_label\", \"disease\", \"study_name\"]],\n",
    "        briney_patients[[\"participant_label\", \"disease\", \"study_name\"]],\n",
    "    ],\n",
    "    axis=0,\n",
    ")\n",
    "participant_df_plus_briney"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "participant_df_plus_briney.to_csv(\n",
    "    config.paths.metadata_dir\n",
    "    / \"generated.external_cohorts.all_bcr.participant_metadata.tsv\",\n",
    "    sep=\"\\t\",\n",
    "    index=None,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "-all",
   "notebook_metadata_filter": "-all"
  },
  "kernelspec": {
   "display_name": "py39-cuda-env",
   "language": "python",
   "name": "py39-cuda-env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
